<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Peter N Robinson">
<meta name="author" content="Justin Reese">
<meta name="author" content="Blessy Antony">
<meta name="author" content="Elena Casiraghi">
<meta name="author" content="David Sahner">
<meta name="author" content="T. M. Murali">
<meta name="author" content="Hannah Blau">

<title>The Researcher’s Guide to N3C - 12&nbsp; Machine Learning</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/upcoming.html" rel="next">
<link href="../chapters/support.html" rel="prev">
<link href="../images/n3c-logo-circle-large-0.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-5F685V8M7X"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-5F685V8M7X', { 'anonymize_ip': true});
</script>
<style>

      .quarto-title-block .quarto-title-banner h1,
      .quarto-title-block .quarto-title-banner h2,
      .quarto-title-block .quarto-title-banner h3,
      .quarto-title-block .quarto-title-banner h4,
      .quarto-title-block .quarto-title-banner h5,
      .quarto-title-block .quarto-title-banner h6
      {
        color: #FCB754;
      }

      .quarto-title-block .quarto-title-banner {
        color: #FCB754;
background-image: url(../images/g2n3c-banner-1.png);
background-size: cover;
      }
</style>


<link rel="stylesheet" href="../assets/font-awesome.css">
<link rel="stylesheet" href="../assets/tables.css">
<meta name="citation_title" content="[[12]{.chapter-number}&nbsp; [Machine Learning]{.chapter-title}]{#sec-ml .quarto-section-identifier}">
<meta name="citation_author" content="Peter N Robinson">
<meta name="citation_author" content="Justin Reese">
<meta name="citation_author" content="Blessy Antony">
<meta name="citation_author" content="Elena Casiraghi">
<meta name="citation_author" content="David Sahner">
<meta name="citation_author" content="T. M. Murali">
<meta name="citation_author" content="Hannah Blau">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=The elements of statistical learning: Data mining, inference, and prediction;,citation_author=Trevor Hastie;,citation_author=Robert Tibshirani;,citation_author=Jerome H Friedman;,citation_author=Jerome H Friedman;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_volume=2;">
<meta name="citation_reference" content="citation_title=Hands-on machine learning with scikit-learn, keras, and TensorFlow;,citation_author=Aurélien Géron;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;">
<meta name="citation_reference" content="citation_title=Deep learning with python;,citation_author=Francois Chollet;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;">
<meta name="citation_reference" content="citation_title=17. A value for n-person games;,citation_author=L. S. Shapley;,citation_publication_date=1953-12;,citation_cover_date=1953-12;,citation_year=1953;,citation_fulltext_html_url=https://doi.org/10.1515/9781400881970-018;,citation_doi=10.1515/9781400881970-018;,citation_inbook_title=Contributions to the theory of games (AM-28), volume II;">
<meta name="citation_reference" content="citation_title=Consistent individualized feature attribution for tree ensembles;,citation_author=Scott M. Lundberg;,citation_author=Gabriel G. Erion;,citation_author=Su-In Lee;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://arxiv.org/abs/1802.03888;,citation_doi=10.48550/ARXIV.1802.03888;,citation_publisher=arXiv;">
<meta name="citation_reference" content="citation_title=Artificial intelligence: Emerging applications in biotechnology and pharma;,citation_author=David Sahner;,citation_author=David C. Spellmeyer;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1016/b978-0-12-815585-1.00028-0;,citation_doi=10.1016/b978-0-12-815585-1.00028-0;,citation_inbook_title=Biotechnology entrepreneurship;">
<meta name="citation_reference" content="citation_title=Data normalization and standardization: A technical report;,citation_author=Peshawa J Muhammad Ali;,citation_author=Rezhna Hassan Faraj;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_fulltext_html_url=http://rgdoi.net/10.13140/RG.2.2.28948.04489;,citation_doi=10.13140/RG.2.2.28948.04489;,citation_publisher=Unpublished;">
<meta name="citation_reference" content="citation_title=A method for comparing multiple imputation techniques: A case study on the u.s. National COVID cohort collaborative;,citation_author=Elena Casiraghi;,citation_author=Rachel Wong;,citation_author=Margaret Hall;,citation_author=Ben Coleman;,citation_author=Marco Notaro;,citation_author=Michael D. Evans;,citation_author=Jena S. Tronieri;,citation_author=Hannah Blau;,citation_author=Bryan Laraway;,citation_author=Tiffany J. Callahan;,citation_author=Lauren E. Chan;,citation_author=Carolyn T. Bramante;,citation_author=John B. Buse;,citation_author=Richard A. Moffitt;,citation_author=Til Stürmer;,citation_author=Steven G. Johnson;,citation_author=Yu Raymond Shao;,citation_author=Justin Reese;,citation_author=Peter N. Robinson;,citation_author=Alberto Paccanaro;,citation_author=Giorgio Valentini;,citation_author=Jared D. Huling;,citation_author=Kenneth J. Wilkins;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.1016/j.jbi.2023.104295;,citation_doi=10.1016/j.jbi.2023.104295;,citation_volume=139;,citation_journal_title=Journal of Biomedical Informatics;,citation_publisher=Elsevier BV;">
<meta name="citation_reference" content="citation_title=Learning from data with structured missingness;,citation_author=Robin Mitra;,citation_author=Sarah F. McGough;,citation_author=Tapabrata Chakraborti;,citation_author=Chris Holmes;,citation_author=Ryan Copping;,citation_author=Niels Hagenbuch;,citation_author=Stefanie Biedermann;,citation_author=Jack Noonan;,citation_author=Brieuc Lehmann;,citation_author=Aditi Shenvi;,citation_author=Xuan Vinh Doan;,citation_author=David Leslie;,citation_author=Ginestra Bianconi;,citation_author=Ruben Sanchez-Garcia;,citation_author=Alisha Davies;,citation_author=Maxine Mackintosh;,citation_author=Eleni-Rosalina Andrinopoulou;,citation_author=Anahid Basiri;,citation_author=Chris Harbron;,citation_author=Ben D. MacArthur;,citation_publication_date=2023-01;,citation_cover_date=2023-01;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.1038/s42256-022-00596-z;,citation_issue=1;,citation_doi=10.1038/s42256-022-00596-z;,citation_volume=5;,citation_journal_title=Nature Machine Intelligence;,citation_publisher=Springer Science; Business Media LLC;">
<meta name="citation_reference" content="citation_title=Assessment of the impact of EHR heterogeneity for clinical research through a case study of silent brain infarction;,citation_author=Sunyang Fu;,citation_author=Lester Y. Leung;,citation_author=Anne-Olivia Raulli;,citation_author=David F. Kallmes;,citation_author=Kristin A. Kinsman;,citation_author=Kristoff B. Nelson;,citation_author=Michael S. Clark;,citation_author=Patrick H. Luetmer;,citation_author=Paul R. Kingsbury;,citation_author=David M. Kent;,citation_author=Hongfang Liu;,citation_publication_date=2020-03;,citation_cover_date=2020-03;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1186/s12911-020-1072-9;,citation_issue=1;,citation_doi=10.1186/s12911-020-1072-9;,citation_volume=20;,citation_journal_title=BMC Medical Informatics and Decision Making;,citation_publisher=Springer Science; Business Media LLC;">
<meta name="citation_reference" content="citation_title=Millions of black people affected by racial bias in health-care algorithms;,citation_author=Heidi Ledford;,citation_publication_date=2019-10;,citation_cover_date=2019-10;,citation_year=2019;,citation_fulltext_html_url=https://doi.org/10.1038/d41586-019-03228-6;,citation_issue=7780;,citation_doi=10.1038/d41586-019-03228-6;,citation_volume=574;,citation_journal_title=Nature;,citation_publisher=Springer Science; Business Media LLC;">
<meta name="citation_reference" content="citation_title=Common pitfalls and recommendations for using machine learning to detect and prognosticate for COVID-19 using chest radiographs and CT scans;,citation_author=Michael Roberts;,citation_author=Derek Driggs;,citation_author=Matthew Thorpe;,citation_author=Julian Gilbey;,citation_author=Michael Yeung;,citation_author=Stephan Ursprung;,citation_author=Angelica I. Aviles-Rivero;,citation_author=Christian Etmann;,citation_author=Cathal McCague;,citation_author=Lucian Beer;,citation_author=Jonathan R. Weir-McCall;,citation_author=Zhongzhao Teng;,citation_author=Effrossyni Gkrania-Klotsas;,citation_author=Alessandro Ruggiero;,citation_author=Anna Korhonen;,citation_author=Emily Jefferson;,citation_author=Emmanuel Ako;,citation_author=Georg Langs;,citation_author=Ghassem Gozaliasl;,citation_author=Guang Yang;,citation_author=Helmut Prosch;,citation_author=Jacobus Preller;,citation_author=Jan Stanczuk;,citation_author=Jing Tang;,citation_author=Johannes Hofmanninger;,citation_author=Judith Babar;,citation_author=Lorena Escudero Sánchez;,citation_author=Muhunthan Thillai;,citation_author=Paula Martin Gonzalez;,citation_author=Philip Teare;,citation_author=Xiaoxiang Zhu;,citation_author=Mishal Patel;,citation_author=Conor Cafolla;,citation_author=Hojjat Azadbakht;,citation_author=Joseph Jacob;,citation_author=Josh Lowe;,citation_author=Kang Zhang;,citation_author=Kyle Bradley;,citation_author=Marcel Wassin;,citation_author=Markus Holzer;,citation_author=Kangyu Ji;,citation_author=Maria Delgado Ortet;,citation_author=Tao Ai;,citation_author=Nicholas Walton;,citation_author=Pietro Lio;,citation_author=Samuel Stranks;,citation_author=Tolou Shadbahr;,citation_author=Weizhe Lin;,citation_author=Yunfei Zha;,citation_author=Zhangming Niu;,citation_author=James H. F. Rudd;,citation_author=Evis Sala;,citation_author=Carola-Bibiane Schönlieb and;,citation_publication_date=2021-03;,citation_cover_date=2021-03;,citation_year=2021;,citation_fulltext_html_url=https://doi.org/10.1038/s42256-021-00307-0;,citation_issue=3;,citation_doi=10.1038/s42256-021-00307-0;,citation_volume=3;,citation_journal_title=Nature Machine Intelligence;,citation_publisher=Springer Science; Business Media LLC;">
<meta name="citation_reference" content="citation_title=Generalisable long COVID subtypes: Findings from the NIH N3C and RECOVER programmes;,citation_author=Justin T. Reese;,citation_author=Hannah Blau;,citation_author=Elena Casiraghi;,citation_author=Timothy Bergquist;,citation_author=Johanna J. Loomba;,citation_author=Tiffany J. Callahan;,citation_author=Bryan Laraway;,citation_author=Corneliu Antonescu;,citation_author=Ben Coleman;,citation_author=Michael Gargano;,citation_author=Kenneth J. Wilkins;,citation_author=Luca Cappelletti;,citation_author=Tommaso Fontana;,citation_author=Nariman Ammar;,citation_author=Blessy Antony;,citation_author=T. M. Murali;,citation_author=J. Harry Caufield;,citation_author=Guy Karlebach;,citation_author=Julie A. McMurry;,citation_author=Andrew Williams;,citation_author=Richard Moffitt;,citation_author=Jineta Banerjee;,citation_author=Anthony E. Solomonides;,citation_author=Hannah Davis;,citation_author=Kristin Kostka;,citation_author=Giorgio Valentini;,citation_author=David Sahner;,citation_author=Christopher G. Chute;,citation_author=Charisse Madlock-Brown;,citation_author=Melissa A. Haendel;,citation_author=Peter N. Robinson;,citation_author=Heidi Spratt;,citation_author=Shyam Visweswaran;,citation_author=Joseph Eugene Flack;,citation_author=Yun Jae Yoo;,citation_author=Davera Gabriel;,citation_author=G. Caleb Alexander;,citation_author=Hemalkumar B. Mehta;,citation_author=Feifan Liu;,citation_author=Robert T. Miller;,citation_author=Rachel Wong;,citation_author=Elaine L. Hill;,citation_author=Lorna E. Thorpe;,citation_author=Jasmin Divers;,citation_publication_date=2023-01;,citation_cover_date=2023-01;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.1016/j.ebiom.2022.104413;,citation_doi=10.1016/j.ebiom.2022.104413;,citation_volume=87;,citation_journal_title=eBioMedicine;,citation_publisher=Elsevier BV;">
<meta name="citation_reference" content="citation_title=Explainable machine learning for early assessment of COVID-19 risk prediction in emergency departments;,citation_author=Elena Casiraghi;,citation_author=Dario Malchiodi;,citation_author=Gabriella Trucco;,citation_author=Marco Frasca;,citation_author=Luca Cappelletti;,citation_author=Tommaso Fontana;,citation_author=Alessandro Andrea Esposito;,citation_author=Emanuele Avola;,citation_author=Alessandro Jachetti;,citation_author=Justin Reese;,citation_author=Alessandro Rizzi;,citation_author=Peter N. Robinson;,citation_author=Giorgio Valentini;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1109/access.2020.3034032;,citation_doi=10.1109/access.2020.3034032;,citation_volume=8;,citation_journal_title=IEEE Access;,citation_publisher=Institute of Electrical; Electronics Engineers (IEEE);">
<meta name="citation_reference" content="citation_title=Reflections on defining a standard for computable expression of scientific knowledge: What teach us yoda can;,citation_author=B. S. Alper;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36654808;,citation_issue=1;,citation_doi=10.1002/lrh2.10312;,citation_issn=2379-6146 (Electronic) 2379-6146 (Linking);,citation_volume=7;,citation_journal_title=Learn Health Syst;">
<meta name="citation_reference" content="citation_title=Testing graphical causal models using the r package &amp;amp;amp;quot;dagitty&amp;quot;;,citation_author=A. Ankan;,citation_author=I. M. N. Wortel;,citation_author=J. Textor;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/33592130;,citation_issue=2;,citation_doi=10.1002/cpz1.45;,citation_volume=1;,citation_journal_title=Curr Protoc;">
<meta name="citation_reference" content="citation_title=Long-term use of immunosuppressive medicines and in-hospital COVID-19 outcomes: A retrospective cohort study using data from the national COVID cohort collaborative;,citation_author=K. M. Andersen;,citation_author=B. A. Bates;,citation_author=E. S. Rashidi;,citation_author=A. L. Olex;,citation_author=R. B. Mannon;,citation_author=R. C. Patel;,citation_author=J. Singh;,citation_author=J. Sun;,citation_author=P. G. Auwaerter;,citation_author=D. K. Ng;,citation_author=J. B. Segal;,citation_author=B. T. Garibaldi;,citation_author=H. B. Mehta;,citation_author=G. C. Alexander;,citation_author=Covid Cohort Collaborative Consortium National;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/34806036;,citation_issue=1;,citation_doi=10.1016/S2665-9913(21)00325-8;,citation_issn=2665-9913 (Electronic) 2665-9913 (Linking);,citation_volume=4;,citation_journal_title=Lancet Rheumatol;">
<meta name="citation_reference" content="citation_title=Higher hospitalization and mortality rates among SARS-CoV-2-infected persons in rural america;,citation_author=Alfred Jerrod Anzalone;,citation_author=Ronald Horswell;,citation_author=Brian M Hendricks;,citation_author=San Chu;,citation_author=William B Hillegass;,citation_author=William H Beasley;,citation_author=Jeremy R Harper;,citation_author=Wesley Kimble;,citation_author=Clifford J Rosen;,citation_author=Lucio Miele;,citation_author=others;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=1;,citation_doi=10.1111/jrh.12689;,citation_volume=39;,citation_journal_title=The Journal of Rural Health;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=The REporting of studies conducted using observational routinely-collected health data (RECORD) statement;,citation_author=E. I. Benchimol;,citation_author=L. Smeeth;,citation_author=A. Guttmann;,citation_author=K. Harron;,citation_author=D. Moher;,citation_author=I. Petersen;,citation_author=H. T. Sorensen;,citation_author=E. Elm;,citation_author=S. M. Langan;,citation_author=Record Working Committee;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/26440803;,citation_issue=10;,citation_doi=10.1371/journal.pmed.1001885;,citation_issn=1549-1676 (Electronic) 1549-1277 (Print) 1549-1277 (Linking);,citation_volume=12;,citation_journal_title=PLoS Med;">
<meta name="citation_reference" content="citation_title=Harmonizing units and values of quantitative data elements in a very large nationally pooled electronic health record (EHR) dataset;,citation_author=Katie R Bradwell;,citation_author=Jacob T Wooldridge;,citation_author=Benjamin Amor;,citation_author=Tellen D Bennett;,citation_author=Adit Anand;,citation_author=Carolyn Bremer;,citation_author=Yun Jae Yoo;,citation_author=Zhenglong Qian;,citation_author=Steven G Johnson;,citation_author=Emily R Pfaff;,citation_author=others;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=7;,citation_doi=10.1093/jamia/ocac054;,citation_volume=29;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Fairness in machine learning: A survey.;,citation_author=S Caton;,citation_author=S Haas;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://arxiv.org/abs/2010.04053;,citation_doi=10.48550/arXiv.2010.0405;,citation_journal_title=arXiv;">
<meta name="citation_reference" content="citation_title=A new method of classifying prognostic comorbidity in longitudinal studies: Development and validation;,citation_author=Mary E. Charlson;,citation_author=Peter Pompei;,citation_author=Kathy L. Ales;,citation_author=C.Ronald MacKenzie;,citation_publication_date=1987;,citation_cover_date=1987;,citation_year=1987;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/0021968187901718;,citation_issue=5;,citation_doi=10.1016/0021-9681(87)90171-8;,citation_issn=0021-9681;,citation_volume=40;,citation_journal_title=Journal of Chronic Diseases;">
<meta name="citation_reference" content="citation_title=Informatics playbook;,citation_author=Chunlei Wu;,citation_author=undefined C2DH;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_publisher=https://playbook.cd2h.org/;">
<meta name="citation_reference" content="citation_title=The geographies of community disaster resilience;,citation_author=SL Cutter;,citation_author=KD Ash;,citation_author=CT. Emrich;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_fulltext_html_url=https://doi.org/10.1016/j.gloenvcha.2014.08.005;,citation_issue=Nov 1;,citation_volume=29;,citation_journal_title=Global environmental change;">
<meta name="citation_reference" content="citation_title=COVID-19 TestNorm: A tool to normalize COVID-19 testing names to LOINC codes;,citation_author=Xiao Dong;,citation_author=Jianfu Li;,citation_author=Ekin Soysal;,citation_author=Jiang Bian;,citation_author=Scott L DuVall;,citation_author=Elizabeth Hanchrow;,citation_author=Hongfang Liu;,citation_author=Kristine E Lynch;,citation_author=Michael Matheny;,citation_author=Karthik Natarajan;,citation_author=others;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=9;,citation_doi=10.1093/jamia/ocaa145;,citation_volume=27;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Real-world evidence for assessing pharmaceutical treatments in the context of COVID-19;,citation_author=J. M. Franklin;,citation_author=K. J. Lin;,citation_author=N. M. Gatto;,citation_author=J. A. Rassen;,citation_author=R. J. Glynn;,citation_author=S. Schneeweiss;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/33529354;,citation_issue=4;,citation_doi=10.1002/cpt.2185;,citation_volume=109;,citation_journal_title=Clin Pharmacol Ther;">
<meta name="citation_reference" content="citation_title=When can nonrandomized studies support valid inference regarding effectiveness or safety of new medical treatments?;,citation_author=J. M. Franklin;,citation_author=R. Platt;,citation_author=N. A. Dreyer;,citation_author=A. J. London;,citation_author=G. E. Simon;,citation_author=J. H. Watanabe;,citation_author=M. Horberg;,citation_author=A. Hernandez;,citation_author=R. M. Califf;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/33826756;,citation_issue=1;,citation_doi=10.1002/cpt.2255;,citation_issn=1532-6535 (Electronic) 0009-9236 (Print) 0009-9236 (Linking);,citation_volume=111;,citation_journal_title=Clin Pharmacol Ther;">
<meta name="citation_reference" content="citation_title=Regression and other stories;,citation_author=undefined Andrew Gelman;,citation_author=Aki Vehtari;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=http://www.cambridge.org/9781107676510;,citation_isbn=978-1107676510;">
<meta name="citation_reference" content="citation_title=Clinical concept value sets and interoperability in health data analytics;,citation_author=Sigfried Gold;,citation_author=Andrea Batch;,citation_author=Robert McClure;,citation_author=Guoqian Jiang;,citation_author=Hadi Kharrazi;,citation_author=Rishi Saripalle;,citation_author=Vojtech Huser;,citation_author=Chunhua Weng;,citation_author=Nancy Roderer;,citation_author=Ana Szarfman;,citation_author=others;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6371254;,citation_volume=2018;,citation_conference_title=AMIA annual symposium proceedings;,citation_conference=American Medical Informatics Association;">
<meta name="citation_reference" content="citation_title=Practices, norms, and aspirations regarding the construction, validation, and reuse of code sets in the analysis of real-world data;,citation_author=Sigfried Gold;,citation_author=Harold Lehmann;,citation_author=Lisa Schilling;,citation_author=Wayne Lutters;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_doi=10.1101/2021.10.14.21264917;,citation_journal_title=medRxiv;,citation_publisher=Cold Spring Harbor Laboratory Press;">
<meta name="citation_reference" content="citation_title=Collider bias undermines our understanding of COVID-19 disease risk and severity;,citation_author=G. J. Griffith;,citation_author=T. T. Morris;,citation_author=M. J. Tudball;,citation_author=A. Herbert;,citation_author=G. Mancano;,citation_author=L. Pike;,citation_author=G. C. Sharp;,citation_author=J. Sterne;,citation_author=T. M. Palmer;,citation_author=G. Davey Smith;,citation_author=K. Tilling;,citation_author=L. Zuccolo;,citation_author=N. M. Davies;,citation_author=G. Hemani;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=1;,citation_doi=10.1038/s41467-020-19478-2;,citation_volume=11;,citation_journal_title=Nature communications;,citation_publisher=Nature Publishing Group UK London;">
<meta name="citation_reference" content="citation_title=The National COVID Cohort Collaborative (N3C): Rationale, design, infrastructure, and deployment;,citation_abstract=Coronavirus disease 2019 (COVID-19) poses societal challenges that require expeditious data and knowledge sharing. Though organizational clinical data are abundant, these are largely inaccessible to outside researchers. Statistical, machine learning, and causal analyses are most successful with large-scale data beyond what is available in any given organization. Here, we introduce the National COVID Cohort Collaborative (N3C), an open science community focused on analyzing patient-level data from many centers.The Clinical and Translational Science Award Program and scientific community created N3C to overcome technical, regulatory, policy, and governance barriers to sharing and harmonizing individual-level clinical data. We developed solutions to extract, aggregate, and harmonize data across organizations and data models, and created a secure data enclave to enable efficient, transparent, and reproducible collaborative analytics.Organized in inclusive workstreams, we created legal agreements and governance for organizations and researchers; data extraction scripts to identify and ingest positive, negative, and possible COVID-19 cases; a data quality assurance and harmonization pipeline to create a single harmonized dataset; population of the secure data enclave with data, machine learning, and statistical analytics tools; dissemination mechanisms; and a synthetic data pilot to democratize data access.The N3C has demonstrated that a multisite collaborative learning health network can overcome barriers to rapidly build a scalable infrastructure incorporating multiorganizational clinical data for COVID-19 analytics. We expect this effort to save lives by enabling rapid collaboration among clinicians, researchers, and data scientists to identify treatments and specialized care and thereby reduce the immediate and long-term impacts of COVID-19.;,citation_author=Melissa A Haendel;,citation_author=Christopher G Chute;,citation_author=Tellen D Bennett;,citation_author=David A Eichmann;,citation_author=Justin Guinney;,citation_author=Warren A Kibbe;,citation_author=Philip R O Payne;,citation_author=Emily R Pfaff;,citation_author=Peter N Robinson;,citation_author=Joel H Saltz;,citation_author=Heidi Spratt;,citation_author=Christine Suver;,citation_author=John Wilbanks;,citation_author=Adam B Wilcox;,citation_author=Andrew E Williams;,citation_author=Chunlei Wu;,citation_author=Clair Blacketer;,citation_author=Robert L Bradford;,citation_author=James J Cimino;,citation_author=Marshall Clark;,citation_author=Evan W Colmenares;,citation_author=Patricia A Francis;,citation_author=Davera Gabriel;,citation_author=Alexis Graves;,citation_author=Raju Hemadri;,citation_author=Stephanie S Hong;,citation_author=George Hripscak;,citation_author=Dazhi Jiao;,citation_author=Jeffrey G Klann;,citation_author=Kristin Kostka;,citation_author=Adam M Lee;,citation_author=Harold P Lehmann;,citation_author=Lora Lingrey;,citation_author=Robert T Miller;,citation_author=Michele Morris;,citation_author=Shawn N Murphy;,citation_author=Karthik Natarajan;,citation_author=Matvey B Palchuk;,citation_author=Usman Sheikh;,citation_author=Harold Solbrig;,citation_author=Shyam Visweswaran;,citation_author=Anita Walden;,citation_author=Kellie M Walters;,citation_author=Griffin M Weber;,citation_author=Xiaohan Tanner Zhang;,citation_author=Richard L Zhu;,citation_author=Benjamin Amor;,citation_author=Andrew T Girvin;,citation_author=Amin Manna;,citation_author=Nabeel Qureshi;,citation_author=Michael G Kurilla;,citation_author=Sam G Michael;,citation_author=Lili M Portilla;,citation_author=Joni L Rutter;,citation_author=Christopher P Austin;,citation_author=Ken R Gersing;,citation_author=undefined N3C Consortium;,citation_publication_date=2020-08;,citation_cover_date=2020-08;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1093/jamia/ocaa196;,citation_issue=3;,citation_doi=10.1093/jamia/ocaa196;,citation_issn=1527-974X;,citation_volume=28;,citation_journal_title=Journal of the American Medical Informatics Association;">
<meta name="citation_reference" content="citation_title=Using big data to emulate a target trial when a randomized trial is not available;,citation_author=M. A. Hernan;,citation_author=J. M. Robins;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/26994063;,citation_issue=8;,citation_doi=10.1093/aje/kwv254;,citation_issn=1476-6256 (Electronic) 0002-9262 (Print) 0002-9262 (Linking);,citation_volume=183;,citation_journal_title=Am J Epidemiol;">
<meta name="citation_reference" content="citation_title=Racial disparities in COVID-19 test positivity among people living with HIV in the united states;,citation_author=J. Y. Islam;,citation_author=V. Madhira;,citation_author=J. Sun;,citation_author=A. Olex;,citation_author=N. Franceschini;,citation_author=G. Kirk;,citation_author=R. Patel;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/35306931;,citation_issue=5;,citation_doi=10.1177/09564624221074468;,citation_issn=1758-1052 (Electronic) 0956-4624 (Print) 0956-4624 (Linking);,citation_volume=33;,citation_journal_title=Int J STD AIDS;">
<meta name="citation_reference" content="citation_title=Comparing population-based risk-stratification model performance using demographic, diagnosis and medication data extracted from outpatient electronic health records versus administrative claims;,citation_author=Hadi Kharrazi;,citation_author=Winnie Chi;,citation_author=Hsien-Yen Chang;,citation_author=Thomas M Richards;,citation_author=Jason M Gallagher;,citation_author=Susan M Knudson;,citation_author=Jonathan P Weiner;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_issue=8;,citation_doi=10.1097/MLR.0000000000000754;,citation_volume=55;,citation_journal_title=Medical care;,citation_publisher=JSTOR;">
<meta name="citation_reference" content="citation_title=Crossing boundaries knowledge, disciplinarities, and interdisciplinarities;,citation_author=Julie Thompson Klein;,citation_publication_date=1996;,citation_cover_date=1996;,citation_year=1996;,citation_fulltext_html_url=https://www.google.com/books/edition/Crossing_Boundaries/bNJvYf3ROPAC;,citation_isbn=978-0813916798;,citation_series_title=Knowledge : Disciplinarity and beyond;">
<meta name="citation_reference" content="citation_title=Inherent trade-offs in the fair determination of risk scores;,citation_author=Jon M. Kleinberg;,citation_author=Sendhil Mullainathan;,citation_author=Manish Raghavan;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=http://arxiv.org/abs/1609.05807;,citation_doi=10.48550/arXiv.1609.05807;,citation_volume=abs/1609.05807;,citation_journal_title=CoRR;">
<meta name="citation_reference" content="citation_title=Literate programming;,citation_author=Donald E. Knuth;,citation_publication_date=1984-05;,citation_cover_date=1984-05;,citation_year=1984;,citation_fulltext_html_url=https://doi.org/10.1093/comjnl/27.2.97;,citation_issue=2;,citation_doi=10.1093/comjnl/27.2.97;,citation_issn=0010-4620;,citation_volume=27;,citation_journal_title=Comput. J.;,citation_publisher=Oxford University Press, Inc.;">
<meta name="citation_reference" content="citation_title=Guidance for a causal comparative effectiveness analysis emulating a target trial based on big real world evidence: When to start statin treatment;,citation_author=F. Kuehne;,citation_author=B. Jahn;,citation_author=A. Conrads-Frank;,citation_author=M. Bundo;,citation_author=M. Arvandi;,citation_author=F. Endel;,citation_author=N. Popper;,citation_author=G. Endel;,citation_author=C. Urach;,citation_author=M. Gyimesi;,citation_author=E. J. Murray;,citation_author=G. Danaei;,citation_author=T. A. Gaziano;,citation_author=A. Pandya;,citation_author=U. Siebert;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/31512926;,citation_issue=12;,citation_doi=10.2217/cer-2018-0103;,citation_issn=2042-6313 (Electronic) 2042-6305 (Linking);,citation_volume=8;,citation_journal_title=J Comp Eff Res;">
<meta name="citation_reference" content="citation_title=Use of recommended real-world methods for electronic health record data analysis has not improved over 10 years;,citation_author=Chenyu Li;,citation_author=Abdulrahman M. Alsheikh;,citation_author=Karen A. Robinson;,citation_author=Harold P. Lehmann;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.medrxiv.org/content/early/2023/06/22/2023.06.21.23291706;,citation_doi=10.1101/2023.06.21.23291706;,citation_journal_title=medRxiv;,citation_publisher=Cold Spring Harbor Laboratory Press;">
<meta name="citation_reference" content="citation_title=Clinical, social, and policy factors in COVID-19 cases and deaths: Methodological considerations for feature selection and modeling in county-level analyses;,citation_author=C. Madlock-Brown;,citation_author=K. Wilkens;,citation_author=N. Weiskopf;,citation_author=N. Cesare;,citation_author=S. Bhattacharyya;,citation_author=N. O. Riches;,citation_author=J. Espinoza;,citation_author=D. Dorr;,citation_author=K. Goetz;,citation_author=J. Phuong;,citation_author=A. Sule;,citation_author=H. Kharrazi;,citation_author=F. Liu;,citation_author=C. Lemon;,citation_author=W. G. Adams;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/35421958;,citation_issue=1;,citation_doi=10.1186/s12889-022-13168-y;,citation_issn=1471-2458 (Electronic) 1471-2458 (Linking);,citation_volume=22;,citation_journal_title=BMC Public Health;">
<meta name="citation_reference" content="citation_title=Correction: Clinical, social, and policy factors in COVID-19 cases and deaths: Methodological considerations for feature selection and modeling in county-level analyses;,citation_author=C. Madlock-Brown;,citation_author=K. Wilkens;,citation_author=N. Weiskopf;,citation_author=N. Cesare;,citation_author=S. Bhattacharyya;,citation_author=N. O. Riches;,citation_author=J. Espinoza;,citation_author=D. Dorr;,citation_author=K. Goetz;,citation_author=J. Phuong;,citation_author=A. Sule;,citation_author=H. Kharrazi;,citation_author=F. Liu;,citation_author=C. Lemon;,citation_author=W. G. Adams;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/35751109;,citation_issue=1;,citation_doi=10.1186/s12889-022-13562-6;,citation_issn=1471-2458 (Electronic) 1471-2458 (Linking);,citation_volume=22;,citation_journal_title=BMC Public Health;">
<meta name="citation_reference" content="citation_title=Use of hydroxychloroquine, remdesivir, and dexamethasone among adults hospitalized with covid-19 in the united states: A retrospective cohort study;,citation_author=Hemalkumar B. Mehta;,citation_author=Huijun An;,citation_author=Kathleen M. Andersen;,citation_author=Omar Mansour;,citation_author=Vithal Madhira;,citation_author=Emaan S. Rashidi;,citation_author=Benjamin Bates;,citation_author=Soko Setoguchi;,citation_author=Corey Joseph;,citation_author=Paul T. Kocis;,citation_author=Richard Moffitt;,citation_author=Tellen D. Bennett;,citation_author=Christopher G. Chute;,citation_author=Brian T. Garibaldi;,citation_author=G. Caleb Alexander;,citation_publication_date=2021-10-01;,citation_cover_date=2021-10-01;,citation_year=2021;,citation_issue=10;,citation_doi=10.7326/M21-0857;,citation_issn=0003-4819;,citation_volume=174;,citation_journal_title=Annals of Internal Medicine;,citation_publisher=American College of Physicians;">
<meta name="citation_reference" content="citation_title=Learning from data with structured missingness;,citation_author=Robin Mitra;,citation_author=Sarah F McGough;,citation_author=Tapabrata Chakraborti;,citation_author=Chris Holmes;,citation_author=Ryan Copping;,citation_author=Niels Hagenbuch;,citation_author=Stefanie Biedermann;,citation_author=Jack Noonan;,citation_author=Brieuc Lehmann;,citation_author=Aditi Shenvi;,citation_author=others;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=1;,citation_doi=10.1038/s42256-022-00596-z;,citation_volume=5;,citation_journal_title=Nature Machine Intelligence;,citation_publisher=Nature Publishing Group UK London;">
<meta name="citation_reference" content="citation_title=Identifying the PECO: A framework for formulating good questions to explore the association of environmental and other exposures with health outcomes;,citation_author=R. L. Morgan;,citation_author=P. Whaley;,citation_author=K. A. Thayer;,citation_author=H. J. Schunemann;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/30166065;,citation_issue=Pt 1;,citation_doi=10.1016/j.envint.2018.07.015;,citation_issn=1873-6750 (Electronic) 0160-4120 (Print) 0160-4120 (Linking);,citation_volume=121;,citation_journal_title=Environ Int;">
<meta name="citation_reference" content="citation_title=Increased stroke severity and mortality in patients with SARS-CoV-2 infection: An analysis from the N3C database;,citation_author=J. A. Narrett;,citation_author=I. Mallawaarachchi;,citation_author=C. M. Aldridge;,citation_author=E. D. Assefa;,citation_author=A. Patel;,citation_author=J. J. Loomba;,citation_author=S. Ratcliffe;,citation_author=O. Sadan;,citation_author=T. Monteith;,citation_author=B. B. Worrall;,citation_author=D. E. Brown;,citation_author=K. C. Johnston;,citation_author=A. M. Southerland;,citation_author=N. C. consortium;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36641948;,citation_issue=3;,citation_doi=10.1016/j.jstrokecerebrovasdis.2023.106987;,citation_issn=1532-8511 (Electronic) 1052-3057 (Print) 1052-3057 (Linking);,citation_volume=32;,citation_journal_title=J Stroke Cerebrovasc Dis;">
<meta name="citation_reference" content="citation_title=The book of OHDSI: Observational health data sciences and informatics;,citation_author=undefined OHDSI;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://ohdsi.github.io/TheBookOfOhdsi/;,citation_isbn=9781088855195;">
<meta name="citation_reference" content="citation_title=Documentation: Code repositories overview;,citation_author=undefined Palantir;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_publisher=https://www.palantir.com/docs/foundry/code-repositories/overview/;">
<meta name="citation_reference" content="citation_title=Identifying who has long COVID in the USA: A machine learning approach using N3C data.;,citation_author=E. R. Pfaff;,citation_author=A. T. Girvin;,citation_author=T. D. Bennett;,citation_author=A. Bhatia;,citation_author=I. M. Brooks;,citation_author=R. R. Deer;,citation_author=J. P. Dekermanjian;,citation_author=S. E. Jolley;,citation_author=M. G. Kahn;,citation_author=K. Kostka;,citation_author=J. A. McMurry;,citation_author=R. Moffitt;,citation_author=A. Walden;,citation_author=C. G. Chute;,citation_author=M. A. Haendel;,citation_author=C. Bramante;,citation_author=D. Dorr;,citation_author=M. Morris;,citation_author=A. M. Parker;,citation_author=H. Sidky;,citation_author=K. Gersing;,citation_author=S. Hong;,citation_author=E. Niehaus;,citation_publication_date=2022-05;,citation_cover_date=2022-05;,citation_year=2022;,citation_fulltext_html_url=https://doi.org/10.1016/S2589-7500(22)00048-6;,citation_issue=7;,citation_doi=10.1016/S2589-7500(22)00048-6;,citation_issn=0010-4620;,citation_volume=4;,citation_journal_title=Lancet Digit Health;">
<meta name="citation_reference" content="citation_title=Synergies between centralized and federated approaches to data quality: A report from the national COVID cohort collaborative;,citation_author=Emily R Pfaff;,citation_author=Andrew T Girvin;,citation_author=Davera L Gabriel;,citation_author=Kristin Kostka;,citation_author=Michele Morris;,citation_author=Matvey B Palchuk;,citation_author=Harold P Lehmann;,citation_author=Benjamin Amor;,citation_author=Mark Bissell;,citation_author=Katie R Bradwell;,citation_author=others;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=4;,citation_doi=10.1093/jamia/ocab217;,citation_volume=29;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Coding long COVID: Characterizing a new disease through an ICD-10 lens;,citation_author=Emily R Pfaff;,citation_author=Charisse Madlock-Brown;,citation_author=John M Baratta;,citation_author=Abhishek Bhatia;,citation_author=Hannah Davis;,citation_author=Andrew Girvin;,citation_author=Elaine Hill;,citation_author=Elizabeth Kelly;,citation_author=Kristin Kostka;,citation_author=Johanna Loomba;,citation_author=others;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=1;,citation_doi=10.1186/s12916-023-02737-6;,citation_volume=21;,citation_journal_title=BMC medicine;,citation_publisher=BioMed Central;">
<meta name="citation_reference" content="citation_title=COVID vaccine hesitancy and risk of a traffic crash;,citation_author=D. A. Redelmeier;,citation_author=J. Wang;,citation_author=D. Thiruchelvam;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36470796;,citation_issue=2;,citation_doi=10.1016/j.amjmed.2022.11.002;,citation_issn=1555-7162 (Electronic) 0002-9343 (Print) 0002-9343 (Linking);,citation_volume=136;,citation_journal_title=Am J Med;">
<meta name="citation_reference" content="citation_title=Generalisable long COVID subtypes: Findings from the NIH N3C and RECOVER programmes;,citation_author=Justin T Reese;,citation_author=Hannah Blau;,citation_author=Elena Casiraghi;,citation_author=Timothy Bergquist;,citation_author=Johanna J Loomba;,citation_author=Tiffany J Callahan;,citation_author=Bryan Laraway;,citation_author=Corneliu Antonescu;,citation_author=Ben Coleman;,citation_author=Michael Gargano;,citation_author=others;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_doi=10.1016/j.ebiom.2022.104413;,citation_volume=87;,citation_journal_title=EBioMedicine;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Electronic health records based phenotyping in next-generation clinical trials: A perspective from the NIH health care systems collaboratory;,citation_author=Rachel L Richesson;,citation_author=W Ed Hammond;,citation_author=Meredith Nahm;,citation_author=Douglas Wixted;,citation_author=Gregory E Simon;,citation_author=Jennifer G Robinson;,citation_author=Alan E Bauck;,citation_author=Denise Cifelli;,citation_author=Michelle M Smerek;,citation_author=John Dickerson;,citation_author=others;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=e2;,citation_doi=10.1136/amiajnl-2013-001926;,citation_volume=20;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=BMJ Publishing Group;">
<meta name="citation_reference" content="citation_title=Graphical depiction of longitudinal study designs in health care databases;,citation_author=S. Schneeweiss;,citation_author=J. A. Rassen;,citation_author=J. S. Brown;,citation_author=K. J. Rothman;,citation_author=L. Happe;,citation_author=P. Arlett;,citation_author=G. Dal Pan;,citation_author=W. Goettsch;,citation_author=W. Murk;,citation_author=S. V. Wang;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/30856654;,citation_issue=6;,citation_doi=10.7326/M18-3079;,citation_issn=1539-3704 (Electronic) 0003-4819 (Linking);,citation_volume=170;,citation_journal_title=Ann Intern Med;">
<meta name="citation_reference" content="citation_title=Improving reproducibility by using high-throughput observational studies with empirical calibration;,citation_author=M. J. Schuemie;,citation_author=P. B. Ryan;,citation_author=G. Hripcsak;,citation_author=D. Madigan;,citation_author=M. A. Suchard;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/30082302;,citation_issue=2128;,citation_doi=10.1098/rsta.2017.0356;,citation_issn=1471-2962 (Electronic) 1364-503X (Print) 1364-503X (Linking);,citation_volume=376;,citation_journal_title=Philos Trans A Math Phys Eng Sci;">
<meta name="citation_reference" content="citation_title=Large-scale evidence generation and evaluation across a network of databases (LEGEND): Assessing validity using hypertension as a case study;,citation_author=M. J. Schuemie;,citation_author=P. B. Ryan;,citation_author=N. Pratt;,citation_author=R. Chen;,citation_author=S. C. You;,citation_author=H. M. Krumholz;,citation_author=D. Madigan;,citation_author=G. Hripcsak;,citation_author=M. A. Suchard;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/32827027;,citation_issue=8;,citation_doi=10.1093/jamia/ocaa124;,citation_issn=1527-974X (Electronic) 1067-5027 (Print) 1067-5027 (Linking);,citation_volume=27;,citation_journal_title=J Am Med Inform Assoc;">
<meta name="citation_reference" content="citation_title=Outcomes of COVID-19 in patients with cancer: Report from the national COVID cohort collaborative (N3C);,citation_author=Noha Sharafeldin;,citation_author=Benjamin Bates;,citation_author=Qianqian Song;,citation_author=Vithal Madhira;,citation_author=Yao Yan;,citation_author=Sharlene Dong;,citation_author=Eileen Lee;,citation_author=Nathaniel Kuhrt;,citation_author=Yu Raymond Shao;,citation_author=Feifan Liu;,citation_author=Timothy Bergquist;,citation_author=Justin Guinney;,citation_author=Jing Su;,citation_author=Umit Topaloglu;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=20;,citation_doi=10.1200/JCO.21.01074;,citation_volume=39;,citation_journal_title=Journal of Clinical Oncology;">
<meta name="citation_reference" content="citation_title=Data quality considerations for evaluating COVID-19 treatments using real world data: Learnings from the national COVID cohort collaborative (N3C);,citation_author=H. Sidky;,citation_author=J. C. Young;,citation_author=A. T. Girvin;,citation_author=E. Lee;,citation_author=Y. R. Shao;,citation_author=N. Hotaling;,citation_author=S. Michael;,citation_author=K. J. Wilkins;,citation_author=S. Setoguchi;,citation_author=M. J. Funk;,citation_author=N. C. Consortium;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36800930;,citation_issue=1;,citation_doi=10.1186/s12874-023-01839-2;,citation_issn=1471-2288 (Electronic) 1471-2288 (Linking);,citation_volume=23;,citation_journal_title=BMC Med Res Methodol;">
<meta name="citation_reference" content="citation_title=Principles for data analysis workflows;,citation_author=S. Stoudt;,citation_author=V. N. Vasquez;,citation_author=C. C. Martinez;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/33735208;,citation_issue=3;,citation_doi=10.1371/journal.pcbi.1008770;,citation_issn=1553-7358 (Electronic) 1553-734X (Print) 1553-734X (Linking);,citation_volume=17;,citation_journal_title=PLoS Comput Biol;">
<meta name="citation_reference" content="citation_title=Association between immune dysfunction and COVID-19 breakthrough infection after SARS-CoV-2 vaccination in the US;,citation_author=Jing Sun;,citation_author=Qulu Zheng;,citation_author=Vithal Madhira;,citation_author=Amy L. Olex;,citation_author=Alfred J. Anzalone;,citation_author=Amanda Vinson;,citation_author=Jasvinder A. Singh;,citation_author=Evan French;,citation_author=Alison G. Abraham;,citation_author=Jomol Mathew;,citation_author=Nasia Safdar;,citation_author=Gaurav Agarwal;,citation_author=Kathryn C. Fitzgerald;,citation_author=Namrata Singh;,citation_author=Umit Topaloglu;,citation_author=Christopher G. Chute;,citation_author=Roslyn B. Mannon;,citation_author=Gregory D. Kirk;,citation_author=Rena C. Patel;,citation_publication_date=2022-02;,citation_cover_date=2022-02;,citation_year=2022;,citation_issue=2;,citation_doi=10.1001/jamainternmed.2021.7024;,citation_issn=2168-6106;,citation_volume=182;,citation_journal_title=Archives of internal medicine (Chicago, Ill. : 1908);,citation_publisher=American Medical Association;">
<meta name="citation_reference" content="citation_title=Informative missingness: What can we learn from patterns in missing laboratory data in the electronic health record?;,citation_author=A. L. M. Tan;,citation_author=E. J. Getzen;,citation_author=M. R. Hutch;,citation_author=Z. H. Strasser;,citation_author=A. Gutierrez-Sacristan;,citation_author=T. T. Le;,citation_author=A. Dagliati;,citation_author=M. Morris;,citation_author=D. A. Hanauer;,citation_author=B. Moal;,citation_author=C. L. Bonzel;,citation_author=W. Yuan;,citation_author=L. Chiudinelli;,citation_author=P. Das;,citation_author=H. G. Zhang;,citation_author=B. J. Aronow;,citation_author=P. Avillach;,citation_author=G. A. Brat;,citation_author=T. Cai;,citation_author=C. Hong;,citation_author=W. G. La Cava;,citation_author=H. Hooi Will Loh;,citation_author=Y. Luo;,citation_author=S. N. Murphy;,citation_author=K. Yuan Hgiam;,citation_author=G. S. Omenn;,citation_author=L. P. Patel;,citation_author=M. Jebathilagam Samayamuthu;,citation_author=E. R. Shriver;,citation_author=Z. Shakeri Hossein Abad;,citation_author=B. W. L. Tan;,citation_author=S. Visweswaran;,citation_author=X. Wang;,citation_author=G. M. Weber;,citation_author=Z. Xia;,citation_author=B. Verdy;,citation_author=Covid- EHR;,citation_author=Q. Long;,citation_author=D. L. Mowery;,citation_author=J. H. Holmes;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36738870;,citation_doi=10.1016/j.jbi.2023.104306;,citation_volume=139;,citation_journal_title=J Biomed Inform;">
<meta name="citation_reference" content="citation_title=Software as a medical device (SAMD): Clinical evaluation/guidance for industry and food and drug administration staff;,citation_author=U.S. Food and Drug Administration;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_fulltext_html_url=https://www.fda.gov/media/100714/download;,citation_publisher=FDA;">
<meta name="citation_reference" content="citation_title=Developing real-world data and evidence to support regulatory decision-making;,citation_author=U.S. Food and Drug Administration and the Duke-Margolis Center for Health Policy;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://www.youtube.com/watch?v=-G6ltatA71I;">
<meta name="citation_reference" content="citation_title=Good machine learning practice for medical device development: Guiding principles;,citation_author=U.S. Food and Drug Administration, Health Canada, and the United Kingdom’s Medicines and Healthcare products Regulatory Agency (MHRA);,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://www.fda.gov/medical-devices/software-medical-device-samd/good-machine-learning-practice-medical-device-development-guiding-principles;,citation_publisher=FDA;">
<meta name="citation_reference" content="citation_title=Considerations for the design and conduct of externally controlled trials for drug and biological products guidance for industry;,citation_author=U.S. Food and Drug Administration;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.fda.gov/media/164960/download;,citation_technical_report_institution=Food; Drug Administration;">
<meta name="citation_reference" content="citation_title=The strengthening the reporting of observational studies in epidemiology (STROBE) statement: Guidelines for reporting observational studies;,citation_author=E. Elm;,citation_author=D. G. Altman;,citation_author=M. Egger;,citation_author=S. J. Pocock;,citation_author=P. C. Gotzsche;,citation_author=J. P. Vandenbroucke;,citation_author=Strobe Initiative;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/25046131;,citation_issue=12;,citation_doi=10.1016/j.ijsu.2014.07.013;,citation_issn=1743-9159 (Electronic) 1743-9159 (Linking);,citation_volume=12;,citation_journal_title=Int J Surg;">
<meta name="citation_reference" content="citation_title=Synthea™ novel coronavirus (COVID-19) model and synthetic data set;,citation_author=Jason Walonoski;,citation_author=Sybil Klaus;,citation_author=Eldesia Granger;,citation_author=Dylan Hall;,citation_author=Andrew Gregorowicz;,citation_author=George Neyarapally;,citation_author=Abigail Watson;,citation_author=Jeff Eastman;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S2666521220300077;,citation_doi=doi.org/10.1016/j.ibmed.2020.100007;,citation_issn=2666-5212;,citation_volume=1-2;,citation_journal_title=Intelligence-Based Medicine;">
<meta name="citation_reference" content="citation_title=STaRT-RWE: Structured template for planning and reporting on the implementation of real world evidence studies;,citation_author=S. V. Wang;,citation_author=S. Pinheiro;,citation_author=W. Hua;,citation_author=P. Arlett;,citation_author=Y. Uyama;,citation_author=J. A. Berlin;,citation_author=D. B. Bartels;,citation_author=K. H. Kahler;,citation_author=L. G. Bessette;,citation_author=S. Schneeweiss;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/33436424;,citation_doi=10.1136/bmj.m4856;,citation_volume=372;,citation_journal_title=BMJ;">
<meta name="citation_reference" content="citation_title=Healthcare utilization is a collider: An introduction to collider bias in EHR data reuse;,citation_author=N. G. Weiskopf;,citation_author=D. A. Dorr;,citation_author=C. Jackson;,citation_author=H. P. Lehmann;,citation_author=C. A. Thompson;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36752649;,citation_doi=10.1093/jamia/ocad013;,citation_journal_title=J Am Med Inform Assoc;">
<meta name="citation_reference" content="citation_title=R for data science;,citation_author=undefined Hadley Wickham;,citation_author=Garrett Grolemund;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://r4ds.hadley.nz/;,citation_isbn=978-1491910399;">
<meta name="citation_reference" content="citation_title=The FAIR guiding principles for scientific data management and stewardship;,citation_author=Mark D Wilkinson;,citation_author=Michel Dumontier;,citation_author=IJsbrand Jan Aalbersberg;,citation_author=Gabrielle Appleton;,citation_author=Myles Axton;,citation_author=Arie Baak;,citation_author=Niklas Blomberg;,citation_author=Jan-Willem Boiten;,citation_author=Luiz Bonino Silva Santos;,citation_author=Philip E Bourne;,citation_author=others;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_doi=10.1038/sdata.2016.18;,citation_volume=3;,citation_journal_title=Scientific data;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Knitr: A comprehensive tool for reproducible research in R;,citation_author=Yihui Xie;,citation_editor=Victoria Stodden;,citation_editor=Friedrich Leisch;,citation_editor=Roger D. Peng;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_fulltext_html_url=http://www.crcpress.com/product/isbn/9781466561595;,citation_isbn=978-1466561595;,citation_inbook_title=Implementing reproducible computational research;">
<meta name="citation_reference" content="citation_title=Dynamic documents with R and knitr;,citation_author=Yihui Xie;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_fulltext_html_url=http://yihui.org/knitr/;,citation_isbn=978-1498716963;">
<meta name="citation_reference" content="citation_title=Bookdown: Authoring books and technical documents with R markdown;,citation_author=Yihui Xie;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://bookdown.org/yihui/bookdown;,citation_isbn=978-1138700109;">
<meta name="citation_reference" content="citation_title=R markdown: The definitive guide;,citation_author=Yihui Xie;,citation_author=J. J. Allaire;,citation_author=Garrett Grolemund;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://bookdown.org/yihui/rmarkdown;,citation_isbn=978-1138359338;">
<meta name="citation_reference" content="citation_title=R markdown cookbook;,citation_author=Yihui Xie;,citation_author=Christophe Dervieux;,citation_author=Emily Riederer;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://bookdown.org/yihui/rmarkdown-cookbook;,citation_isbn=9780367563837;">
<meta name="citation_reference" content="citation_title=Associations between HIV infection and clinical spectrum of COVID-19: A population level analysis based on US national COVID cohort collaborative (N3C) data;,citation_author=Xueying Yang;,citation_author=Jing Sun;,citation_author=Rena C Patel;,citation_author=Jiajia Zhang;,citation_author=Siyuan Guo;,citation_author=Qulu Zheng;,citation_author=Amy L Olex;,citation_author=Bankole Olatosi;,citation_author=Sharon B Weissman;,citation_author=Jessica Y Islam;,citation_author=others;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=11;,citation_doi=10.1016/S2352-3018(21)00239-3;,citation_volume=8;,citation_journal_title=The Lancet HIV;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Comparative effectiveness of dexamethasone in treatment of hospitalized COVID-19 patients during the first year of the pandemic: The N3C data repository;,citation_author=R. Zhou;,citation_author=K. E. Johnson;,citation_author=J. F. Rousseau;,citation_author=P. J. Rathouz;,citation_author=N. C. Consortium;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pubmed/36324806;,citation_doi=10.1101/2022.10.22.22281373;,citation_journal_title=medRxiv;">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/support.html">Special Topics</a></li><li class="breadcrumb-item"><a href="../chapters/ml.html"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Machine Learning</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/support.html">Special Topics</a></li><li class="breadcrumb-item"><a href="../chapters/ml.html"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Machine Learning</span></a></li></ol></nav>
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title"><span id="sec-ml" class="quarto-section-identifier"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></h1>
                      </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Authors</div>
    <div class="quarto-title-meta-heading">Affiliations</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Peter N Robinson <a href="mailto:peter.robinson@bih-charite.de" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0002-0736-9199" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://www.bihealth.org/">
              Berlin Institute of Health at Charité
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">Justin Reese <a href="mailto:justinreese@lbl.gov" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0002-2170-2250" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://biosciences.lbl.gov/profiles/justin-reese/">
              Lawrence Berkeley National Laboratory
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">Blessy Antony <a href="mailto:blessyantony@vt.edu" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0002-1269-5191" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://cs.vt.edu/">
              Virginia Tech
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">Elena Casiraghi <a href="mailto:elena.casiraghi@unimi.com" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0003-2024-7572" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://casiraghi.di.unimi.it/">
              Università degli Studi di Milano
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">David Sahner <a href="mailto:david.sahner@nih.gov" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0002-4616-1171" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://ncats.nih.gov/">
              National Center for Advancing Translational Sciences
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">T. M. Murali <a href="mailto:murali@cs.vt.edu" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0003-3688-4672" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://bioinformatics.cs.vt.edu/~murali/">
              Virginia Tech
              </a>
            </p>
        </div>
      <div class="quarto-title-meta-contents">
      <p class="author">Hannah Blau <a href="mailto:hannah.blau@jax.org" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0003-4557-5492" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://www.jax.org/">
              The Jackson Laboratory for Genomic Medicine
              </a>
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">The Researcher’s Guide to N3C</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="../The-Researcher-s-Guide-to-N3C.epub" title="Download ePub" class="quarto-navigation-tool px-1" aria-label="Download ePub"><i class="bi bi-journal"></i></a>
    <div class="dropdown">
      <a href="" title="Share" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Share"><i class="bi bi-share"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
              <i class="bi bi-bi-twitter pe-1"></i>
            Twitter
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
              <i class="bi bi-bi-facebook pe-1"></i>
            Facebook
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Getting Started</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/story.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A Research Story</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Life Cycle - From Patients to N3C Researchers</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/governance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Governance, Leadership, and Operations Structures</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/onboarding.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Onboarding, Enclave Access, N3C Team Science</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Investigating</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/access.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Getting &amp; Managing Data Access</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/understanding.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Understanding the Data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/tools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Introducing Enclave Analysis Tools</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/practices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Best Practices for the Research Life Cycle</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/publishing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Publishing and Sharing Your Work</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Special Topics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Help and Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/ml.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/upcoming.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Upcoming Topics</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Back Matter</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/funding.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Funding and Institutional Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">References</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-ml-introduction" id="toc-sec-ml-introduction" class="nav-link active" data-scroll-target="#sec-ml-introduction"><span class="header-section-number">12.1</span> Introduction</a></li>
  <li><a href="#sec-ml-gap" id="toc-sec-ml-gap" class="nav-link" data-scroll-target="#sec-ml-gap"><span class="header-section-number">12.2</span> Good Algorithmic Practice (GAP)</a></li>
  <li><a href="#sec-ml-in-n3c" id="toc-sec-ml-in-n3c" class="nav-link" data-scroll-target="#sec-ml-in-n3c"><span class="header-section-number">12.3</span> ML in N3C</a>
  <ul class="collapse">
  <li><a href="#sec-languages-and-libraries" id="toc-sec-languages-and-libraries" class="nav-link" data-scroll-target="#sec-languages-and-libraries"><span class="header-section-number">12.3.1</span> Languages and libraries available in N3C</a></li>
  <li><a href="#sec-example-project" id="toc-sec-example-project" class="nav-link" data-scroll-target="#sec-example-project"><span class="header-section-number">12.3.2</span> Example project</a></li>
  </ul></li>
  <li><a href="#other-topics" id="toc-other-topics" class="nav-link" data-scroll-target="#other-topics"><span class="header-section-number">12.4</span> Other topics</a>
  <ul class="collapse">
  <li><a href="#deploying-this-ml-code-to-your-projectdur" id="toc-deploying-this-ml-code-to-your-projectdur" class="nav-link" data-scroll-target="#deploying-this-ml-code-to-your-projectdur"><span class="header-section-number">12.4.1</span> Deploying this ML code to your project/DUR</a></li>
  <li><a href="#using-spark-ml-pyspark.ml-for-time-and-compute-intensive-tasks" id="toc-using-spark-ml-pyspark.ml-for-time-and-compute-intensive-tasks" class="nav-link" data-scroll-target="#using-spark-ml-pyspark.ml-for-time-and-compute-intensive-tasks"><span class="header-section-number">12.4.2</span> Using Spark ML pyspark.ml for time and compute intensive tasks</a></li>
  <li><a href="#using-r-in-code-workbooks" id="toc-using-r-in-code-workbooks" class="nav-link" data-scroll-target="#using-r-in-code-workbooks"><span class="header-section-number">12.4.3</span> Using R in code workbooks</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/edit/main/chapters/ml.md" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/blob/main/chapters/ml.md" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<p><strong>Chapter Leads:</strong> Peter Robinson, Justin Reese</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>This material for this chapter is mostly complete, and we’re adding the final touches before officially publishing it.</p>
<p>In the meantime, please enjoy this pre-release version.</p>
<p>If you have comments, please file a <a href="https://github.com/National-COVID-Cohort-Collaborative/guide-to-n3c-v1/issues">New Issue</a> and tag <a href="https://github.com/chrisroederucdenver">Chris Roeder</a> (GitHub username is <code>@chrisroederucdenver</code>).</p>
</div>
</div>
<section id="sec-ml-introduction" class="level2" data-number="12.1">
<h2 data-number="12.1" class="anchored" data-anchor-id="sec-ml-introduction"><span class="header-section-number">12.1</span> Introduction</h2>
<p>Machine learning (ML) is a method of teaching computers to learn a classification or regression function from data, or detect patterns (e.g., clusters) in data, without explicitly programming the model. Machine learning is a powerful tool for solving complex problems, and it is widely used in biomedicine and many other fields.</p>
<p>For biomedical applications, two main types of ML algorithms are generally used: supervised and unsupervised. In supervised learning, the labels (i.e., true classifications) for the input data used to train the model (e.g., which patients have a disease of interest, or develop an outcome of interest) are known. The goal of supervised learning is to learn a mapping function that can accurately predict the label for new, unseen input examples. A practical example of classification is reported in one of the first publications by the <a href="../chapters/onboarding.html#dt">ML Domain Team</a>, in which the aim was to classify COVID-patients based on their prognosis <span class="citation" data-cites="casiraghi_2020">(<a href="../references.html#ref-casiraghi_2020" role="doc-biblioref">2020</a>)</span>.</p>
<p>In unsupervised ML, models are trained on input data without labels. In this case, the algorithm aims to discover structure within the data in order to facilitate the assignment of labels. One of the most widely known unsupervised ML techniques is clustering, which aims at partitioning the input cases into classes, where each class corresponds to similar cases according to a similarity/dissimilarity metric. In the clinical/medical field, clustering is often used to identify subtypes of a disease of interest. A practical example of clustering was also recently published by our ML group on the identification of long-COVID subtypes <span class="citation" data-cites="reese_2023">(<a href="../references.html#ref-reese_2023" role="doc-biblioref">2023</a>)</span>.</p>
<p>The intention of this chapter is to demonstrate how to perform ML in N3C by walking the reader through a specific ML use case with working code.</p>
</section>
<section id="sec-ml-gap" class="level2" data-number="12.2">
<h2 data-number="12.2" class="anchored" data-anchor-id="sec-ml-gap"><span class="header-section-number">12.2</span> Good Algorithmic Practice (GAP)</h2>
<p>Knowledge of GAP is crucial to the successful application of machine learning to biomedical data. We intend to demonstrate GAP in this chapter by example. A full discussion of this is beyond the scope of this document, but a partial list of important considerations, with references for further reading, is as follows:</p>
<ul>
<li><strong>Generalizability</strong> – models trained on one set of data often do not perform well on data originating from other places (e.g., other hospital systems) or other times (e.g., in a different period of time in the COVID pandemic) <span class="citation" data-cites="roberts_2021">(<a href="../references.html#ref-roberts_2021" role="doc-biblioref">2021</a>)</span>. Since the goal of supervised machine learning is to develop generalizable algorithms that perform well on completely unseen (i.e., independent) test data, practitioners must be familiar with various techniques and considerations that mitigate the risk of failure to generalize (i.e., model overfitting) in the appropriate context, such as regularization, dimensionality reduction, feature extraction, cross-validation, and use of a model type that is appropriate given the amount of available data (e.g., deep learning algorithms may require relatively large training sets) <span class="citation" data-cites="chollet_2021">Géron (<a href="../references.html#ref-geron_2022" role="doc-biblioref">2022</a>)</span>. The machine learning practitioner should thoroughly understand the potential trade-off between under-fitting and over-fitting a predictive model <span class="citation" data-cites="hastie_2009">(<a href="../references.html#ref-hastie_2009" role="doc-biblioref">2009</a>)</span>. In addition, N3C data is not a random sample, and it worth familiarizing yourself with the data as described in Chapters <a href="understanding.html" class="quarto-xref"><span>7</span></a> and <a href="cycle.html" class="quarto-xref"><span>3</span></a>.</li>
<li><strong>Social bias</strong> – scientific literature abounds with ML models and results that contain bias with respect to factors such as gender, ethnicity, and race. It is critical to realize that the choice of data or ML model may reinforce harmful bias in scientific literature <span class="citation" data-cites="roberts_2021">(<a href="../references.html#ref-roberts_2021" role="doc-biblioref">2021</a>)</span>. This can happen even if you do not intend to introduce such a bias.</li>
<li><strong>Data Considerations</strong>
<ul>
<li><strong>Site-Level Heterogeneity</strong> – N3C contains data from a large number of different hospital/healthcare systems (identified by the variable <code>data_parter_id</code> in the N3C <code>person</code> table) with several distinct common data models (ACT, OMOP, PCORNET, TRINETX). Features can vary wildly between data partners due to factors such as differing data models, patient populations, coding practices, and ETL code required for harmonization of data. This can negatively affect ML models, for example by limiting generalizability of models that depend on features that differ between sites <span class="citation" data-cites="fu_2020">(<a href="../references.html#ref-fu_2020" role="doc-biblioref">2020</a>)</span>. It is thus helpful to understand the N3C data ingestion and harmonization processes described in Chapter <a href="cycle.html" class="quarto-xref"><span>3</span></a>.</li>
<li><strong>Missingness</strong> – Your data will likely have missingness, sometimes quite often. How you deal with this can greatly affect your ML results. <span class="citation" data-cites="mitra_2023">(<a href="../references.html#ref-mitra_2023" role="doc-biblioref">2023</a>)</span> See also Casiraghi, et al.<span class="citation" data-cites="casiraghi_2023">(<a href="../references.html#ref-casiraghi_2023" role="doc-biblioref">2023</a>)</span> for a detailed description of missingness specifically in N3C, and strategies for dealing with it, such as imputation. Ultimately, how you deal with missingness is subject to how reasonable your assumptions are concerning relationships between observed data and desired-yet-missing information, assumptions best guided by engaging with subject matter experts familiar with the data relationships at hand (see below).</li>
<li><strong>Features for ML</strong> – In the example use case we describe here, we will leverage the N3C Logic Liaison COVID-19 Diagnosed or Lab Confirmed Patients fact table as a source of features. If this table does not contain the data you need for your ML task, you can explore other available datasets or use the Logic Liaison templates to construct additional derived variables, or create your own custom derived variables. Appropriate data pre-processing, feature selection and/or feature engineering is a crucial step in the machine learning process. Depending upon the type of algorithm, data may also need to be normalized to reduce the risk of excessive influence of features in certain classifiers <span class="citation" data-cites="muhammad_2014 sahner_2020">(<a href="../references.html#ref-muhammad_2014" role="doc-biblioref">2014</a>; <a href="../references.html#ref-sahner_2020" role="doc-biblioref">2020</a>)</span>.</li>
</ul></li>
<li><strong>Engaging with subject matter experts</strong> – Your ML project will greatly benefit from subject matter experts to guide your experimental design and to help interpret your results. If you are performing ML on data from people with, for example, diabetes, you should engage at least one clinical diabetes expert with whom you can collaborate. One of the strengths of N3C is the large and diverse community of scientists, and it is likely that there is an expert in almost any field of interest who can help. Subject matter experts can provide important help in feature selection, assessments of windows for variables, choice of feature transformations and interaction terms in a model, and definition of model performance goals (sensitivity versus specificity trade-off). N3C <a href="../chapters/onboarding.html#sec-onboarding-dts">Domain Teams</a> are an excellent source of such expertise, as many are organized around clinical topics such as Diabetes, Kidney Injury, Pregnancy, and many more.</li>
<li><strong>Explainability</strong> – Of great interest to practitioners is the explainability of machine learning models, as deep learning tools that rely upon progressive abstractions of data in hidden layers may seem like “black boxes,” especially to those who are not versed in machine learning. Tools are available to help assess feature importance (e.g., SHAP values <span class="citation" data-cites="lundburg_2018">(<a href="../references.html#ref-lundburg_2018" role="doc-biblioref">2018</a>)</span> <span class="citation" data-cites="shapley_1953">(<a href="../references.html#ref-shapley_1953" role="doc-biblioref">1953</a>)</span></li>
<li><strong>Adequate Documentation</strong> – Clearly annotated code, with explicit characterization of methodology and techniques that are employed, as well as descriptions of all key steps in a pipeline, including hyperparameter choice/search, appropriate train/test splits, etc., are critical to reproducibility of research. In addition, The N3C Enclave provides a feature called Protocol Pad which can explain exactly how a study was conducted in N3C.</li>
</ul>
</section>
<section id="sec-ml-in-n3c" class="level2" data-number="12.3">
<h2 data-number="12.3" class="anchored" data-anchor-id="sec-ml-in-n3c"><span class="header-section-number">12.3</span> ML in N3C</h2>
<p>Developing ML pipelines in the N3C platform is similar to developing code to solve other tasks, e.g., statistical analyses. The main challenges involve preparing the data to form cohorts, selecting features, and understanding how to develop your code.</p>
<section id="sec-languages-and-libraries" class="level3" data-number="12.3.1">
<h3 data-number="12.3.1" class="anchored" data-anchor-id="sec-languages-and-libraries"><span class="header-section-number">12.3.1</span> Languages and libraries available in N3C</h3>
<p>ML in N3C is typically performed using Python or R. Pipelines are also sometimes mixed-language (R, Python, SQL, Java). Most major ML libraries in Python and R are already installed and additional libraries can be made available on a project-specific basis. The examples in this chapter will use Python. For long-running and memory-intensive machine learning jobs, pyspark provides the library <a href="https://spark.apache.org/docs/2.2.0/api/python/pyspark.ml.html">pyspark.ml</a>.</p>
<section id="sec-what-you-should-know" class="level4" data-number="12.3.1.1">
<h4 data-number="12.3.1.1" class="anchored" data-anchor-id="sec-what-you-should-know"><span class="header-section-number">12.3.1.1</span> What you should know before reading this chapter:</h4>
<ul>
<li>Understanding of GAP (see GAP section above)</li>
<li>Basic proficiency in software development</li>
<li>Basic proficiency in Python (see <a href="https://www.python.org/about/gettingstarted/">Python.org website</a> for some references on learning Python)</li>
<li>Basic knowledge of ML (see <a href="https://www.oreilly.com/library/view/hands-on-machine-learning/9781098125967/">Hands–on Machine Learning with Scikit–Learn, Keras, and TensorFlow</a> for a good reference)</li>
<li>Basic knowledge of EHR data, especially data that follows the OMOP common data model (see the <a href="https://ohdsi.github.io/TheBookOfOhdsi/">Book of OHDSI</a> for a good reference)</li>
<li>Basic familiarity with code repositories (see <a href="../chapters/tools.html#sec-tools-apps-repo">Code Repositories</a> for N3C documentation)</li>
<li>Basic familiarity with code workbooks (see <a href="../chapters/tools.html#sec-tools-apps-workbook">Code Workbooks</a> for N3C documentation)</li>
</ul>
</section>
</section>
<section id="sec-example-project" class="level3" data-number="12.3.2">
<h3 data-number="12.3.2" class="anchored" data-anchor-id="sec-example-project"><span class="header-section-number">12.3.2</span> Example project</h3>
<p>To demonstrate how to implement ML software in N3C, we will walk through a specific example: predicting cardiovascular sequelae (specifically, atrial fibrillation/flutter) from acute COVID.</p>
<section id="sec-strategy" class="level4" data-number="12.3.2.1">
<h4 data-number="12.3.2.1" class="anchored" data-anchor-id="sec-strategy"><span class="header-section-number">12.3.2.1</span> Strategy</h4>
<ol type="1">
<li><strong>Create a reusable ML library package.</strong> This library contains the implementations of standard machine learning classification algorithms such as logistic regression, naïve Bayes, random forest, support vector machine, and multi-layer perceptron. This package can be imported and used for any downstream classification applications.</li>
<li><strong>Define cohort and features using the N3C Logic Liaison COVID-19 Diagnosed or Lab Confirmed Patients fact table</strong> (See also Chapter <a href="tools.html" class="quarto-xref"><span>8</span></a>.) This table provides many variables (including records for COVID-associated conditions such as diabetes or acute kidney injury by the timing of each: before, during and after acute COVID-19 infection) for each patient and is useful for ML tasks.</li>
<li><strong>Choose one post-COVID indicator column as a label to predict.</strong> Here we will predict cardiovascular sequelae following acute COVID. In this example, we will use ‘Atrial Fibrillation Flutter (AFF)’ post COVID-19 as an indicator for cardiovascular sequelae.</li>
<li><strong>Predict this label using a single ML model.</strong> We will use the random forest model from the ML library created in step 1 to predict AFF post COVID. One can use as an alternative any of the classification models available in the library.</li>
</ol>
</section>
<section id="sec-walkthrough" class="level4" data-number="12.3.2.2">
<h4 data-number="12.3.2.2" class="anchored" data-anchor-id="sec-walkthrough"><span class="header-section-number">12.3.2.2</span> Walkthrough:</h4>
<p>We implemented the above strategy using code repositories. Code repositories greatly facilitate software development in the Enclave. Useful features include a debugger, a GitHub-like workflow (pull requests, branches, diffs, etc.) to coordinate among developers, and a convenient method to export code to GitHub. In addition, the process for specifying Python packages to be used is much more convenient, and (in the authors’ experience) code repositories are also more stable and less prone to runtime issues than code workbooks. Two notable limitations of code repositories should be noted:</p>
<ol type="1">
<li>Plots/visualization must be done separately in a code workbook (as in step 3 below) and</li>
<li>The R language is not supported in code repositories as of July 2023, so performing statistical analyses with R packages also requires a code workbook.</li>
</ol>
<p>For our walkthrough, we will be creating and using a Python library repository <a href="https://github.com/National-COVID-Cohort-Collaborative/ml-classification-pipeline">ml-classification-pipeline Python library</a> and a data transform code repository that Blessy Antony (Virginia Tech, blessyantony@vt.edu) developed to do machine learning in N3C.</p>
<p>For the basics of creating a code repository and using its debugger, see Chapter <a href="tools.html" class="quarto-xref"><span>8</span></a>.</p>
<ol type="1">
<li><p>Reusable ML library package</p>
<ol type="a">
<li><p><strong>Create a folder in a workspace.</strong> Go to your DUR workspace or “Practice Area - Public and Example Data” Left sidebar → Search → “Practice Area - Public and Example Data” → click on first hit → make a new folder</p>
<div id="fig-ml-01-create-folder-in-workspace" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Create a folder in your workspace of choice">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-01-create-folder-in-workspace-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-01-create-folder-in-workspace.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" data-glightbox="description: .lightbox-desc-1"><img src="images/ml/fig-ml-01-create-folder-in-workspace.png" class="img-fluid figure-img" alt="Create a folder in your workspace of choice"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-01-create-folder-in-workspace-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.1: Create a folder in your workspace of choice (private DUR or public practice area) where all your code will be located.
</figcaption>
</figure>
</div></li>
<li><p><strong>Create a Python library code repository to implement the ML algorithms.</strong> New → code repository → Python Library → initialize repository</p>
<div id="fig-ml-02-create-python-lib-code-repo" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Create a python library code repository">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-02-create-python-lib-code-repo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-02-create-python-lib-code-repo.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" data-glightbox="description: .lightbox-desc-2"><img src="images/ml/fig-ml-02-create-python-lib-code-repo.png" class="img-fluid figure-img" alt="Create a python library code repository"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-02-create-python-lib-code-repo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.2: Create a python library code repository that will contain all reusable code to be later used in other downstream code repositories.
</figcaption>
</figure>
</div>
<p>This repository provides two frameworks that implement logistic regression, random forest, support vector machine, and multi-layer perceptron - scikit-learn and pyspark.ml at ml-classification-pipeline/src/models/.</p></li>
<li><p><strong>Adding dependencies to code repository</strong> – See the N3C documentation on discovering and using Python libraries in code repositories.</p></li>
<li><p><strong>Implementation of classification algorithms</strong> – The scikit-learn and pyspark implementations have the following code organization -</p>
<ol type="i">
<li><p><strong>dataset_preprocessor.py</strong> – Methods to drop column(s) from the input dataset, split the dataset into training and test datasets based on a given ratio, sample the training and/or testing datasets to have approximately equal number of positive and negative samples (if required), standardize the datasets.</p>
<p>The pyspark implementation contains an additional method to assemble the input into libsvm format.</p></li>
<li><p><strong>classification_model.py</strong> – The ClassificationModel class implements a classification pipeline. First, preprocess the dataset and compute training and testing datasets using the methods in dataset_preprocessor.py. Second, perform hyperparameter search and train classification models using k-fold cross validation. Third, evaluate the trained models on test datasets. Finally, perform multiple iterations of these three steps to generalize the results.</p></li>
<li><p><strong>classification algorithms</strong> – The logistic_regression.py, multi_layer_perceptron.py, random_forest.py, and svm.py files implement the respective classification algorithms with model instantiation and definition of the hyperparameter search. The classes defined in each of these files extend the ClassificationModel class defined in classification_model.py and leverage the pipeline implemented in the parent class.</p></li>
</ol></li>
</ol></li>
<li><p>Post COVID cardiovascular sequelae prediction <!-- https://serial-comma.com/blog/posts/2020-09-13-hanging-paragraphs-in-markdown.html --></p>
<ol type="a">
<li><p><strong>Create a Python transforms code repository.</strong> Go to the folder that you had created in step 1a and create a new python transforms code repository - New → code repository → Data transforms (Python) → initialize repository:</p>
<div id="fig-ml-03-create-python-data-transforms-code-repo" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Create a python data transforms code repository">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-03-create-python-data-transforms-code-repo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-03-create-python-data-transforms-code-repo.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" data-glightbox="description: .lightbox-desc-3"><img src="images/ml/fig-ml-03-create-python-data-transforms-code-repo.png" class="img-fluid figure-img" alt="Create a python data transforms code repository"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-03-create-python-data-transforms-code-repo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.3: Create a python data transforms code repository for cardiovascular sequelae prediction that will use the library created in step 1.
</figcaption>
</figure>
</div></li>
<li><p><strong>Define cohort and features using COVID patient facts table.</strong> Create a new transform called “aff_create_cohort.py” Left sidebar → select folder where you want to create the file → right click → New File → Enter filename → Select ‘Python Transformation (*.py)’ from the drop down → Create</p>
<div id="fig-ml-04-create-python-transformation-file" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Create a new python transformation script file within a code repository.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-04-create-python-transformation-file-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-04-create-python-transformation-file.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" data-glightbox="description: .lightbox-desc-4"><img src="images/ml/fig-ml-04-create-python-transformation-file.png" class="img-fluid figure-img" alt="Create a new python transformation script file within a code repository."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-04-create-python-transformation-file-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.4: Create a new python transformation script file within a code repository.
</figcaption>
</figure>
</div>
<p>We’ve created a code repository, now we can create a transform to build a random forest model using the ml-classification-pipeline library (note that this library can also be used in code workbooks):</p></li>
<li><p><strong>Add the ml-classification-pipeline library.</strong> We’ve created a code repository, now we can create a transform to build a random forest model using the ml-classification-pipeline library (note that this library can also be used in code workbooks). First, we need to add the library, by navigating to the Sidebar -&gt; Libraries -&gt; search for ml-classification-pipeline.<br><br></p>
<div id="fig-ml-05-search-custom-library" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Search for a custom-created python library in the code repository.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-05-search-custom-library-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-05-search-custom-library.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" data-glightbox="description: .lightbox-desc-5"><img src="images/ml/fig-ml-05-search-custom-library.png" class="img-fluid figure-img" alt="Search for a custom-created python library in the code repository."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-05-search-custom-library-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.5: Search for a custom-created python library in the code repository.
</figcaption>
</figure>
</div>
<p><br><br>Next, add the library by clicking “Add and install library”:<br></p>
<div id="fig-ml-06-add-custom-library" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Add a custom-created python library in the code repository.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-06-add-custom-library-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-06-add-custom-library.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" data-glightbox="description: .lightbox-desc-6"><img src="images/ml/fig-ml-06-add-custom-library.png" class="img-fluid figure-img" alt="Add a custom-created python library in the code repository."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-06-add-custom-library-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.6: Add a custom-created python library in the code repository.
</figcaption>
</figure>
</div>
<p><br><br>NOTE: An error message similar to the one below is sometimes encountered during this step. N3C support may be necessary to resolve this error, see Chapter <a href="support.html" class="quarto-xref"><span>11</span></a>.<br><br></p>
<div id="fig-ml-07-n3c-access-error-custom-library" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Access error while adding custom-created libraries in code repositories.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-07-n3c-access-error-custom-library-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-07-n3c-access-error-custom-library.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" data-glightbox="description: .lightbox-desc-7"><img src="images/ml/fig-ml-07-n3c-access-error-custom-library.png" class="img-fluid figure-img" alt="Access error while adding custom-created libraries in code repositories."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-07-n3c-access-error-custom-library-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.7: Access error while adding custom-created libraries in code repositories. N3C support may be necessary to resolve this error.
</figcaption>
</figure>
</div>
<p><br><br>Next, go back to the ‘files’ tab on the left sidebar, and commit your changes.</p></li>
<li><p><strong>Implement a random forest model for the prediction task.</strong> Create a new transform that trains a random forest on this cohort:</p>
<div id="fig-ml-08-create-python-script-file" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Create python script file.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-08-create-python-script-file-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-08-create-python-script-file.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" data-glightbox="description: .lightbox-desc-8"><img src="images/ml/fig-ml-08-create-python-script-file.png" class="img-fluid figure-img" alt="Create python script file."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-08-create-python-script-file-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.8: Create python script file.
</figcaption>
</figure>
</div>
<p>Now some code - this uses the output of <code>aff_create_cohort.py</code> as input, and runs a random forest on it. You’ll need to change this line to be a path to somewhere you can write a dataset:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>allpatients_output<span class="op">=</span>Output(<span class="st">"PATH_TO_YOUR_OUTPUT_DATASET"</span>),</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>And change this line to the path or the Resource Identifier (RID) of the dataset you saved above in <code>aff_create_cohort.py</code>:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>allpatients_df<span class="op">=</span>Input(<span class="st">"PUT IN PATH OR RID FOR OUTPUT OF aff_create_cohort.py"</span>),</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> transforms.api <span class="im">import</span> transform, configure, Input, Output</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> models.sklearn <span class="im">import</span> random_forest</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="at">@configure</span>(profile<span class="op">=</span>[<span class="st">'DRIVER_MEMORY_EXTRA_LARGE'</span>])</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="at">@transform</span>(</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  allpatients_output<span class="op">=</span>Output(<span class="st">"PATH_TO_YOUR_OUTPUT_DATASET"</span>),</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  allpatients_df<span class="op">=</span>Input(<span class="st">"PUT IN PATH OR RID FOR OUTPUT OF aff_create_cohort.py"</span>),</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> compute(allpatients_df, allpatients_output):</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    n_iterations <span class="op">=</span> <span class="dv">5</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    train_test_ratio <span class="op">=</span> [<span class="fl">0.8</span>, <span class="fl">0.2</span>]</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    preprocess_exclude_columns <span class="op">=</span> [<span class="st">"COVID_associated_hospitalization_indicator"</span>,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"COVID_index_date"</span>]</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    standardize_exclude_columns <span class="op">=</span> [<span class="st">"person_id"</span>, <span class="st">"label"</span>]</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    allpatients_df <span class="op">=</span> allpatients_df.dataframe()</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    allpatients_output.write_dataframe(run(allpatients_df,</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>                                           n_iterations,</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>                                           train_test_ratio,</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>                                           preprocess_exclude_columns,</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>                                           standardize_exclude_columns))</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> run(dataset,</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>        n_iterations,</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>        train_test_ratio,</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>        preprocess_exclude_columns,</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>        standardize_exclude_columns):</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>     random_forest_model <span class="op">=</span> random_forest.RandomForestModel(dataset,</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>                                                           n_iterations,</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>                                                           train_test_ratio,</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>                                                           preprocess_exclude_columns,</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>                                                           standardize_exclude_columns)</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>     <span class="cf">return</span> random_forest_model.run()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ol></li>
<li><p><strong>Visualize the results.</strong> Code repositories do not support creating and viewing visualizations. Hence, we will use code workbooks to create visualizations.</p>
<ol type="a">
<li><p><strong>Create a code workbook.</strong> Navigate to the folder location where you want to create the code workbook → Right Click → New →Code Workbook.</p>
<div id="fig-ml-09-create-codeworkbook" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Create a new code workbook.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-09-create-codeworkbook-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-09-create-codeworkbook.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" data-glightbox="description: .lightbox-desc-9"><img src="images/ml/fig-ml-09-create-codeworkbook.png" class="img-fluid figure-img" alt="Create a new code workbook."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-09-create-codeworkbook-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.9: Create a new code workbook.
</figcaption>
</figure>
</div>
<p>Additional details about the usage, customization, and features of Code Workbooks are discussed in Chapter <a href="tools.html" class="quarto-xref"><span>8</span></a>.</p></li>
<li><p><strong>Compute metrics to evaluate prediction models.</strong> Create a new Python transform node: New transform(top left) → Python code The name of the node is the name of the function to be implemented within the node and it is also the name of the output dataset generated by this node.</p>
<p>Nodes are the atomic units of a code workbook in which we implement individual functions.</p>
<p>Each function takes in one or more input datasets and produces one output dataset. We can import datasets into code workbooks. When the output of a node is provided as an input to another node within the same code workbook, a dependency edge is established from the former to the latter node.</p>
<div id="fig-ml-10-referencing-nodes-codeworkbook" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Referencing nodes in a code workbook.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-10-referencing-nodes-codeworkbook-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-10-referencing-nodes-codeworkbook.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10" data-glightbox="description: .lightbox-desc-10"><img src="images/ml/fig-ml-10-referencing-nodes-codeworkbook.png" class="img-fluid figure-img" alt="Referencing nodes in a code workbook."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-10-referencing-nodes-codeworkbook-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.10: Referencing nodes in a code workbook.
</figcaption>
</figure>
</div>
<p><br>Example: Implement the computation of area under precision recall curve (AUPRC) for the predictions of Logistic Regression models.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AUPRC Computation</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> logistic_regression_allpatients_auprc(<span class="st">"REFERENCE TO THE LOGISTIC REGRESSION OUTPUT DATASET (e.g. aff_allpatients_lr_output)"</span>):</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="im">from</span> sklearn.metrics <span class="im">import</span> precision_recall_curve, auc</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    <span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> aff_allpatients_lr_output</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    iterations <span class="op">=</span> df.run.unique()</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    model_name <span class="op">=</span> <span class="st">"Logistic Regression"</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    cohort_name <span class="op">=</span> <span class="st">"All Patients"</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>    auprc_list <span class="op">=</span> []</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> itr <span class="kw">in</span> iterations:</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>        df_itr <span class="op">=</span> df[df[<span class="st">"run"</span>] <span class="op">==</span> itr]</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>        y_test <span class="op">=</span> df_itr.loc[:,<span class="st">"test_label"</span>].values</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>        y_pred <span class="op">=</span> df_itr.loc[:,<span class="st">"test_prediction"</span>].values</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>        precision, recall, thresholds <span class="op">=</span> precision_recall_curve(y_test, y_pred)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>        auprc <span class="op">=</span> auc(recall, precision)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>        auprc_list.append({<span class="st">"auprc"</span>: auprc,</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"run"</span>: itr,</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"model"</span>: model_name,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"cohort"</span>: cohort_name})</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> pd.DataFrame(auprc_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Similarly, create multiple copies of this node with the same code but different inputs and outputs to generate AUPRC scores for Random Forest, SVM,&nbsp; and MLP models.</p>
<p><strong>Templates</strong></p>
<p>To overcome the cumbersome task of creating copies of the same piece of code in multiple nodes to perform the same function over different inputs (for example, computation of AUPRC scores for the outputs of different prediction models), we can create a reusable template which allows parameterization of the inputs for nodes.</p>
<p>More details and examples about <a href="https://unite.nih.gov/docs/foundry/code-workbook/templates-getting-started">creation and usage of templates is found in the N3C documentation</a> <i class="fa-solid fa-lock" title="Link requires an N3C Data Enclave account" aria-label="lock"></i>. More information about Code Workbooks and Templates may be found in Section <a href="tools.html#sec-tools-store-templates" class="quarto-xref"><span>8.3.2</span></a>.</p></li>
<li><p><strong>Create visualizations.</strong> Create another node in the workbook to draw Box plots of the AUPRC scores of the Logistic Regression, Random Forest, SVM, MLP models computed for ten iterations of the classification of cardiovascular sequelae. The inputs to this node will be the outputs generated from the nodes in the previous step that computed the AUPRC scores for each of the prediction methods.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># AUPRC Distribution Boxplot Visualization</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> allpatients_auprc_boxplot(logistic_regression_allpatients_auprc,</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>                              svm_allpatients_auprc,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                              mlp_allpatients_auprc,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                              rf_allpatients_auprc):</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> pd.concat([logistic_regression_allpatients_auprc,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                    svm_allpatients_auprc,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>                    mlp_allpatients_auprc,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>                    rf_allpatients_auprc])</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> sns.boxplot(x<span class="op">=</span><span class="st">"model"</span>, y<span class="op">=</span><span class="st">"auprc"</span>, data<span class="op">=</span>df, notch<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">"Area under Precision-Recall Curve : All Patients"</span>)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Select the ‘visualization’ tab at the bottom to view the output of this node.<br></p>
<div id="fig-ml-11-viewing-visualization-outputs" class="quarto-figure quarto-figure-center quarto-float anchored" alt="Viewing visualization outputs in Code Workbook.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ml-11-viewing-visualization-outputs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="images/ml/fig-ml-11-viewing-visualization-outputs.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11" data-glightbox="description: .lightbox-desc-11"><img src="images/ml/fig-ml-11-viewing-visualization-outputs.png" class="img-fluid figure-img" alt="Viewing visualization outputs in Code Workbook."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ml-11-viewing-visualization-outputs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12.11: Viewing visualization outputs in Code Workbook.
</figcaption>
</figure>
</div></li>
</ol></li>
</ol>
</section>
</section>
</section>
<section id="other-topics" class="level2" data-number="12.4">
<h2 data-number="12.4" class="anchored" data-anchor-id="other-topics"><span class="header-section-number">12.4</span> Other topics</h2>
<section id="deploying-this-ml-code-to-your-projectdur" class="level3" data-number="12.4.1">
<h3 data-number="12.4.1" class="anchored" data-anchor-id="deploying-this-ml-code-to-your-projectdur"><span class="header-section-number">12.4.1</span> Deploying this ML code to your project/DUR</h3>
<p>You can use the code above in your project by following these steps:</p>
<ol type="1">
<li><p>Open the code repository containing the code you would like to copy</p></li>
<li><p>Open the pulldown menu by the “Clone” button</p></li>
<li><p>Copy the git remote URL to your clipboard</p></li>
<li><p>On your local machine, open a command line and run this command:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> clone [paste the remote URL]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
<li><p>In your DUR, create an empty code repository</p></li>
<li><p>In the empty code repository, open the pulldown menu by the “Clone” button and copy the git remote URL to your clipboard as above in steps (2) and (3)</p></li>
<li><p>In the cloned repository on your local machine, run these commands: See <a href="publishing.html#sec-publishing-tech-code-github">Pulling Changes, Pushing to GitHub, Branches and Code Repositories</a> for details on how to do this. <br></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> remote add mynewrepo [paste remote URL of new repository]</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> push mynewrepo master</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div></li>
</ol>
</section>
<section id="using-spark-ml-pyspark.ml-for-time-and-compute-intensive-tasks" class="level3" data-number="12.4.2">
<h3 data-number="12.4.2" class="anchored" data-anchor-id="using-spark-ml-pyspark.ml-for-time-and-compute-intensive-tasks"><span class="header-section-number">12.4.2</span> Using Spark ML pyspark.ml for time and compute intensive tasks</h3>
<p>The examples above use standard Python to train and apply ML models. For ML tasks that are more compute intensive, consider using Apache Spark’s MLlib ML/pyspark.ml instead. A full discussion of Spark ML is beyond the scope of this chapter, but an implementation of a Spark ML model is provided in the same codebase as above, under the module named pysparkml. The official MLlib guide to SparkML is a good resource, and there are specific integrations with the Enclave documented here.</p>
</section>
<section id="using-r-in-code-workbooks" class="level3" data-number="12.4.3">
<h3 data-number="12.4.3" class="anchored" data-anchor-id="using-r-in-code-workbooks"><span class="header-section-number">12.4.3</span> Using R in code workbooks</h3>
<p>In the examples code reported above, we reported Python code, which works within Code Repositories (see <a href="tools.html#sec-tools-apps-repo" class="quarto-xref"><span>Section 8.4.4</span></a>). However, some ML algorithms make use of statistical inference/univariate statistics or multiple-imputation techniques, whose more reliable implementation (with underlying C/C++ engine) is available in R. If you prefer to use R, you can use code within Code Workbooks (see <a href="tools.html#sec-tools-apps-workbook" class="quarto-xref"><span>Section 8.4.3</span></a>). Note also that the Enclave very recently enabled support for R Studio in the Enclave. Finally, a useful “Enclave ML Resources Overview” with links to various resources is available in the <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-3D8340A&amp;view=focus">N3C Knowledge Store</a>.</p>


<div class="hidden" aria-hidden="true">
<span class="glightbox-desc lightbox-desc-1">Figure&nbsp;12.1: Create a folder in your workspace of choice (private DUR or public practice area) where all your code will be located.</span>
<span class="glightbox-desc lightbox-desc-2">Figure&nbsp;12.2: Create a python library code repository that will contain all reusable code to be later used in other downstream code repositories.</span>
<span class="glightbox-desc lightbox-desc-3">Figure&nbsp;12.3: Create a python data transforms code repository for cardiovascular sequelae prediction that will use the library created in step 1.</span>
<span class="glightbox-desc lightbox-desc-4">Figure&nbsp;12.4: Create a new python transformation script file within a code repository.</span>
<span class="glightbox-desc lightbox-desc-5">Figure&nbsp;12.5: Search for a custom-created python library in the code repository.</span>
<span class="glightbox-desc lightbox-desc-6">Figure&nbsp;12.6: Add a custom-created python library in the code repository.</span>
<span class="glightbox-desc lightbox-desc-7">Figure&nbsp;12.7: Access error while adding custom-created libraries in code repositories. N3C support may be necessary to resolve this error.</span>
<span class="glightbox-desc lightbox-desc-8">Figure&nbsp;12.8: Create python script file.</span>
<span class="glightbox-desc lightbox-desc-9">Figure&nbsp;12.9: Create a new code workbook.</span>
<span class="glightbox-desc lightbox-desc-10">Figure&nbsp;12.10: Referencing nodes in a code workbook.</span>
<span class="glightbox-desc lightbox-desc-11">Figure&nbsp;12.11: Viewing visualization outputs in Code Workbook.</span>
</div>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list" style="display: none">
<div id="ref-casiraghi_2020" class="csl-entry" role="listitem">
Casiraghi, E., Malchiodi, D., Trucco, G., Frasca, M., Cappelletti, L., Fontana, T., Esposito, A. A., Avola, E., Jachetti, A., Reese, J., Rizzi, A., Robinson, P. N., &amp; Valentini, G. (2020). Explainable machine learning for early assessment of <span>COVID</span>-19 risk prediction in emergency departments. <em><span>IEEE</span> Access</em>, <em>8</em>, 196299–196325. <a href="https://doi.org/10.1109/access.2020.3034032">https://doi.org/10.1109/access.2020.3034032</a>
</div>
<div id="ref-casiraghi_2023" class="csl-entry" role="listitem">
Casiraghi, E., Wong, R., Hall, M., Coleman, B., Notaro, M., Evans, M. D., Tronieri, J. S., Blau, H., Laraway, B., Callahan, T. J., Chan, L. E., Bramante, C. T., Buse, J. B., Moffitt, R. A., Stürmer, T., Johnson, S. G., Shao, Y. R., Reese, J., Robinson, P. N., … Wilkins, K. J. (2023). A method for comparing multiple imputation techniques: A case study on the u.s. National <span>COVID</span> cohort collaborative. <em>Journal of Biomedical Informatics</em>, <em>139</em>, 104295. <a href="https://doi.org/10.1016/j.jbi.2023.104295">https://doi.org/10.1016/j.jbi.2023.104295</a>
</div>
<div id="ref-chollet_2021" class="csl-entry" role="listitem">
Chollet, F. (2021). <em>Deep learning with python</em>. Simon; Schuster.
</div>
<div id="ref-fu_2020" class="csl-entry" role="listitem">
Fu, S., Leung, L. Y., Raulli, A.-O., Kallmes, D. F., Kinsman, K. A., Nelson, K. B., Clark, M. S., Luetmer, P. H., Kingsbury, P. R., Kent, D. M., &amp; Liu, H. (2020). Assessment of the impact of <span>EHR</span> heterogeneity for clinical research through a case study of silent brain infarction. <em><span>BMC</span> Medical Informatics and Decision Making</em>, <em>20</em>(1). <a href="https://doi.org/10.1186/s12911-020-1072-9">https://doi.org/10.1186/s12911-020-1072-9</a>
</div>
<div id="ref-geron_2022" class="csl-entry" role="listitem">
Géron, A. (2022). <em>Hands-on machine learning with scikit-learn, keras, and TensorFlow</em>. " O’Reilly Media, Inc.".
</div>
<div id="ref-hastie_2009" class="csl-entry" role="listitem">
Hastie, T., Tibshirani, R., Friedman, J. H., &amp; Friedman, J. H. (2009). <em>The elements of statistical learning: Data mining, inference, and prediction</em> (Vol. 2). Springer.
</div>
<div id="ref-lundburg_2018" class="csl-entry" role="listitem">
Lundberg, S. M., Erion, G. G., &amp; Lee, S.-I. (2018). <em>Consistent individualized feature attribution for tree ensembles</em>. arXiv. <a href="https://doi.org/10.48550/ARXIV.1802.03888">https://doi.org/10.48550/ARXIV.1802.03888</a>
</div>
<div id="ref-mitra_2023" class="csl-entry" role="listitem">
Mitra, R., McGough, S. F., Chakraborti, T., Holmes, C., Copping, R., Hagenbuch, N., Biedermann, S., Noonan, J., Lehmann, B., Shenvi, A., et al. (2023). Learning from data with structured missingness. <em>Nature Machine Intelligence</em>, <em>5</em>(1), 13–23. <a href="https://doi.org/10.1038/s42256-022-00596-z">https://doi.org/10.1038/s42256-022-00596-z</a>
</div>
<div id="ref-muhammad_2014" class="csl-entry" role="listitem">
Peshawa J Muhammad Ali, &amp; Rezhna Hassan Faraj. (2014). <em>Data normalization and standardization: A technical report</em>. <a href="https://doi.org/10.13140/RG.2.2.28948.04489">https://doi.org/10.13140/RG.2.2.28948.04489</a>
</div>
<div id="ref-reese_2023" class="csl-entry" role="listitem">
Reese, J. T., Blau, H., Casiraghi, E., Bergquist, T., Loomba, J. J., Callahan, T. J., Laraway, B., Antonescu, C., Coleman, B., Gargano, M., et al. (2023). Generalisable long COVID subtypes: Findings from the NIH N3C and RECOVER programmes. <em>EBioMedicine</em>, <em>87</em>. <a href="https://doi.org/10.1016/j.ebiom.2022.104413">https://doi.org/10.1016/j.ebiom.2022.104413</a>
</div>
<div id="ref-roberts_2021" class="csl-entry" role="listitem">
Roberts, M., Driggs, D., Thorpe, M., Gilbey, J., Yeung, M., Ursprung, S., Aviles-Rivero, A. I., Etmann, C., McCague, C., Beer, L., Weir-McCall, J. R., Teng, Z., Gkrania-Klotsas, E., Ruggiero, A., Korhonen, A., Jefferson, E., Ako, E., Langs, G., Gozaliasl, G., … and, C.-B. S. (2021). Common pitfalls and recommendations for using machine learning to detect and prognosticate for <span>COVID</span>-19 using chest radiographs and <span>CT</span> scans. <em>Nature Machine Intelligence</em>, <em>3</em>(3), 199–217. <a href="https://doi.org/10.1038/s42256-021-00307-0">https://doi.org/10.1038/s42256-021-00307-0</a>
</div>
<div id="ref-sahner_2020" class="csl-entry" role="listitem">
Sahner, D., &amp; Spellmeyer, D. C. (2020). Artificial intelligence: Emerging applications in biotechnology and pharma. In <em>Biotechnology entrepreneurship</em> (pp. 399–417). Elsevier. <a href="https://doi.org/10.1016/b978-0-12-815585-1.00028-0">https://doi.org/10.1016/b978-0-12-815585-1.00028-0</a>
</div>
<div id="ref-shapley_1953" class="csl-entry" role="listitem">
Shapley, L. S. (1953). 17. A value for n-person games. In <em>Contributions to the theory of games (<span>AM</span>-28), volume <span>II</span></em> (pp. 307–318). Princeton University Press. <a href="https://doi.org/10.1515/9781400881970-018">https://doi.org/10.1515/9781400881970-018</a>
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
          // target, if specified
          link.setAttribute("target", "_blank");
          if (link.getAttribute("rel") === null) {
            link.setAttribute("rel", "noopener");
          }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/support.html" class="pagination-link" aria-label="Help and Support">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Help and Support</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/upcoming.html" class="pagination-link" aria-label="Upcoming Topics">
        <span class="nav-page-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Upcoming Topics</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p><a href="https://ncats.nih.gov/n3c/">N3C</a> © <a href="https://creativecommons.org/licenses/by-nd/4.0/"> CC BY-ND 4</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/edit/main/chapters/ml.md" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li><li><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/blob/main/chapters/ml.md" class="toc-action"><i class="bi empty"></i>View source</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Part of <a href="https://covid.cd2h.org/">NIH NCATS</a>.</p>
</div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"openEffect":"zoom","closeEffect":"zoom","selector":".lightbox","loop":false,"descPosition":"bottom"});
window.onload = () => {
  lightboxQuarto.on('slide_before_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    const href = trigger.getAttribute('href');
    if (href !== null) {
      const imgEl = window.document.querySelector(`a[href="${href}"] img`);
      if (imgEl !== null) {
        const srcAttr = imgEl.getAttribute("src");
        if (srcAttr && srcAttr.startsWith("data:")) {
          slideConfig.href = srcAttr;
        }
      }
    } 
  });

  lightboxQuarto.on('slide_after_load', (data) => {
    const { slideIndex, slideNode, slideConfig, player, trigger } = data;
    if (window.Quarto?.typesetMath) {
      window.Quarto.typesetMath(slideNode);
    }
  });

};
          </script>




</body></html>