<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.340">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Harold P. Lehmann">
<meta name="author" content="Lisa Eskenazi">
<meta name="author" content="Shawn O’Neil">
<meta name="author" content="Thomas Richards">
<meta name="author" content="Kristin Kostka">
<meta name="author" content="Sigfried Gold">

<title>The Researcher’s Guide to N3C - 7&nbsp; Understanding the Data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/tools.html" rel="next">
<link href="../chapters/access.html" rel="prev">
<link href="../images/n3c-logo-circle-large-0.png" rel="icon" type="image/png">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-5F685V8M7X"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-5F685V8M7X', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../assets/font-awesome.css">
<link rel="stylesheet" href="../assets/tables.css">
<meta name="citation_title" content="[[7]{.chapter-number}&nbsp; [Understanding the Data]{.chapter-title}]{#sec-understanding .quarto-section-identifier}">
<meta name="citation_author" content="Harold P. Lehmann">
<meta name="citation_author" content="Lisa Eskenazi">
<meta name="citation_author" content="Shawn O&amp;amp;#039;Neil">
<meta name="citation_author" content="Thomas Richards">
<meta name="citation_author" content="Kristin Kostka">
<meta name="citation_author" content="Sigfried Gold">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=Higher hospitalization and mortality rates among SARS-CoV-2-infected persons in rural america;,citation_author=Alfred Jerrod Anzalone;,citation_author=Ronald Horswell;,citation_author=Brian M Hendricks;,citation_author=San Chu;,citation_author=William B Hillegass;,citation_author=William H Beasley;,citation_author=Jeremy R Harper;,citation_author=Wesley Kimble;,citation_author=Clifford J Rosen;,citation_author=Lucio Miele;,citation_author=others;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=1;,citation_doi=10.1111/jrh.12689;,citation_volume=39;,citation_journal_title=The Journal of Rural Health;,citation_publisher=Wiley Online Library;">
<meta name="citation_reference" content="citation_title=Harmonizing units and values of quantitative data elements in a very large nationally pooled electronic health record (EHR) dataset;,citation_author=Katie R Bradwell;,citation_author=Jacob T Wooldridge;,citation_author=Benjamin Amor;,citation_author=Tellen D Bennett;,citation_author=Adit Anand;,citation_author=Carolyn Bremer;,citation_author=Yun Jae Yoo;,citation_author=Zhenglong Qian;,citation_author=Steven G Johnson;,citation_author=Emily R Pfaff;,citation_author=others;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=7;,citation_doi=10.1093/jamia/ocac054;,citation_volume=29;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=A new method of classifying prognostic comorbidity in longitudinal studies: Development and validation;,citation_author=Mary E. Charlson;,citation_author=Peter Pompei;,citation_author=Kathy L. Ales;,citation_author=C.Ronald MacKenzie;,citation_publication_date=1987;,citation_cover_date=1987;,citation_year=1987;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/0021968187901718;,citation_issue=5;,citation_doi=10.1016/0021-9681(87)90171-8;,citation_issn=0021-9681;,citation_volume=40;,citation_journal_title=Journal of Chronic Diseases;">
<meta name="citation_reference" content="citation_title=Informatics playbook;,citation_author=Chunlei Wu;,citation_author=undefined C2DH;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_publisher=https://playbook.cd2h.org/;">
<meta name="citation_reference" content="citation_title=COVID-19 TestNorm: A tool to normalize COVID-19 testing names to LOINC codes;,citation_author=Xiao Dong;,citation_author=Jianfu Li;,citation_author=Ekin Soysal;,citation_author=Jiang Bian;,citation_author=Scott L DuVall;,citation_author=Elizabeth Hanchrow;,citation_author=Hongfang Liu;,citation_author=Kristine E Lynch;,citation_author=Michael Matheny;,citation_author=Karthik Natarajan;,citation_author=others;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_issue=9;,citation_doi=10.1093/jamia/ocaa145;,citation_volume=27;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Regression and other stories;,citation_author=undefined Andrew Gelman;,citation_author=Aki Vehtari;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=http://www.cambridge.org/9781107676510;">
<meta name="citation_reference" content="citation_title=Clinical concept value sets and interoperability in health data analytics;,citation_author=Sigfried Gold;,citation_author=Andrea Batch;,citation_author=Robert McClure;,citation_author=Guoqian Jiang;,citation_author=Hadi Kharrazi;,citation_author=Rishi Saripalle;,citation_author=Vojtech Huser;,citation_author=Chunhua Weng;,citation_author=Nancy Roderer;,citation_author=Ana Szarfman;,citation_author=others;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_volume=2018;,citation_conference_title=AMIA annual symposium proceedings;,citation_conference=American Medical Informatics Association;">
<meta name="citation_reference" content="citation_title=Practices, norms, and aspirations regarding the construction, validation, and reuse of code sets in the analysis of real-world data;,citation_author=Sigfried Gold;,citation_author=Harold Lehmann;,citation_author=Lisa Schilling;,citation_author=Wayne Lutters;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_journal_title=medRxiv;,citation_publisher=Cold Spring Harbor Laboratory Press;">
<meta name="citation_reference" content="citation_title=The National COVID Cohort Collaborative (N3C): Rationale, design, infrastructure, and deployment;,citation_abstract=Coronavirus disease 2019 (COVID-19) poses societal challenges that require expeditious data and knowledge sharing. Though organizational clinical data are abundant, these are largely inaccessible to outside researchers. Statistical, machine learning, and causal analyses are most successful with large-scale data beyond what is available in any given organization. Here, we introduce the National COVID Cohort Collaborative (N3C), an open science community focused on analyzing patient-level data from many centers.The Clinical and Translational Science Award Program and scientific community created N3C to overcome technical, regulatory, policy, and governance barriers to sharing and harmonizing individual-level clinical data. We developed solutions to extract, aggregate, and harmonize data across organizations and data models, and created a secure data enclave to enable efficient, transparent, and reproducible collaborative analytics.Organized in inclusive workstreams, we created legal agreements and governance for organizations and researchers; data extraction scripts to identify and ingest positive, negative, and possible COVID-19 cases; a data quality assurance and harmonization pipeline to create a single harmonized dataset; population of the secure data enclave with data, machine learning, and statistical analytics tools; dissemination mechanisms; and a synthetic data pilot to democratize data access.The N3C has demonstrated that a multisite collaborative learning health network can overcome barriers to rapidly build a scalable infrastructure incorporating multiorganizational clinical data for COVID-19 analytics. We expect this effort to save lives by enabling rapid collaboration among clinicians, researchers, and data scientists to identify treatments and specialized care and thereby reduce the immediate and long-term impacts of COVID-19.;,citation_author=Melissa A Haendel;,citation_author=Christopher G Chute;,citation_author=Tellen D Bennett;,citation_author=David A Eichmann;,citation_author=Justin Guinney;,citation_author=Warren A Kibbe;,citation_author=Philip R O Payne;,citation_author=Emily R Pfaff;,citation_author=Peter N Robinson;,citation_author=Joel H Saltz;,citation_author=Heidi Spratt;,citation_author=Christine Suver;,citation_author=John Wilbanks;,citation_author=Adam B Wilcox;,citation_author=Andrew E Williams;,citation_author=Chunlei Wu;,citation_author=Clair Blacketer;,citation_author=Robert L Bradford;,citation_author=James J Cimino;,citation_author=Marshall Clark;,citation_author=Evan W Colmenares;,citation_author=Patricia A Francis;,citation_author=Davera Gabriel;,citation_author=Alexis Graves;,citation_author=Raju Hemadri;,citation_author=Stephanie S Hong;,citation_author=George Hripscak;,citation_author=Dazhi Jiao;,citation_author=Jeffrey G Klann;,citation_author=Kristin Kostka;,citation_author=Adam M Lee;,citation_author=Harold P Lehmann;,citation_author=Lora Lingrey;,citation_author=Robert T Miller;,citation_author=Michele Morris;,citation_author=Shawn N Murphy;,citation_author=Karthik Natarajan;,citation_author=Matvey B Palchuk;,citation_author=Usman Sheikh;,citation_author=Harold Solbrig;,citation_author=Shyam Visweswaran;,citation_author=Anita Walden;,citation_author=Kellie M Walters;,citation_author=Griffin M Weber;,citation_author=Xiaohan Tanner Zhang;,citation_author=Richard L Zhu;,citation_author=Benjamin Amor;,citation_author=Andrew T Girvin;,citation_author=Amin Manna;,citation_author=Nabeel Qureshi;,citation_author=Michael G Kurilla;,citation_author=Sam G Michael;,citation_author=Lili M Portilla;,citation_author=Joni L Rutter;,citation_author=Christopher P Austin;,citation_author=Ken R Gersing;,citation_author=undefined N3C Consortium;,citation_publication_date=2020-08;,citation_cover_date=2020-08;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1093/jamia/ocaa196;,citation_issue=3;,citation_doi=10.1093/jamia/ocaa196;,citation_issn=1527-974X;,citation_volume=28;,citation_journal_title=Journal of the American Medical Informatics Association;">
<meta name="citation_reference" content="citation_title=Comparing population-based risk-stratification model performance using demographic, diagnosis and medication data extracted from outpatient electronic health records versus administrative claims;,citation_author=Hadi Kharrazi;,citation_author=Winnie Chi;,citation_author=Hsien-Yen Chang;,citation_author=Thomas M Richards;,citation_author=Jason M Gallagher;,citation_author=Susan M Knudson;,citation_author=Jonathan P Weiner;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_issue=8;,citation_doi=10.1097/MLR.0000000000000754;,citation_volume=55;,citation_journal_title=Medical care;,citation_publisher=JSTOR;">
<meta name="citation_reference" content="citation_title=Literate programming;,citation_author=Donald E. Knuth;,citation_publication_date=1984-05;,citation_cover_date=1984-05;,citation_year=1984;,citation_fulltext_html_url=https://doi.org/10.1093/comjnl/27.2.97;,citation_issue=2;,citation_doi=10.1093/comjnl/27.2.97;,citation_issn=0010-4620;,citation_volume=27;,citation_journal_title=Comput. J.;,citation_publisher=Oxford University Press, Inc.;">
<meta name="citation_reference" content="citation_title=Use of hydroxychloroquine, remdesivir, and dexamethasone among adults hospitalized with covid-19 in the united states: A retrospective cohort study;,citation_author=Hemalkumar B. Mehta;,citation_author=Huijun An;,citation_author=Kathleen M. Andersen;,citation_author=Omar Mansour;,citation_author=Vithal Madhira;,citation_author=Emaan S. Rashidi;,citation_author=Benjamin Bates;,citation_author=Soko Setoguchi;,citation_author=Corey Joseph;,citation_author=Paul T. Kocis;,citation_author=Richard Moffitt;,citation_author=Tellen D. Bennett;,citation_author=Christopher G. Chute;,citation_author=Brian T. Garibaldi;,citation_author=G. Caleb Alexander;,citation_publication_date=2021-10-01;,citation_cover_date=2021-10-01;,citation_year=2021;,citation_issue=10;,citation_doi=10.7326/M21-0857;,citation_issn=0003-4819;,citation_volume=174;,citation_journal_title=Annals of Internal Medicine;,citation_publisher=American College of Physicians;">
<meta name="citation_reference" content="citation_title=The book of OHDSI: Observational health data sciences and informatics;,citation_author=undefined OHDSI;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://ohdsi.github.io/TheBookOfOhdsi/;,citation_isbn=9781088855195;">
<meta name="citation_reference" content="citation_title=Documentation: Code repositories overview;,citation_author=undefined Palantir;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_publisher=https://www.palantir.com/docs/foundry/code-repositories/overview/;">
<meta name="citation_reference" content="citation_title=Identifying who has long COVID in the USA: A machine learning approach using N3C data.;,citation_author=E. R. Pfaff;,citation_author=A. T. Girvin;,citation_author=T. D. Bennett;,citation_author=A. Bhatia;,citation_author=I. M. Brooks;,citation_author=R. R. Deer;,citation_author=J. P. Dekermanjian;,citation_author=S. E. Jolley;,citation_author=M. G. Kahn;,citation_author=K. Kostka;,citation_author=J. A. McMurry;,citation_author=R. Moffitt;,citation_author=A. Walden;,citation_author=C. G. Chute;,citation_author=M. A. Haendel;,citation_author=C. Bramante;,citation_author=D. Dorr;,citation_author=M. Morris;,citation_author=A. M. Parker;,citation_author=H. Sidky;,citation_author=K. Gersing;,citation_author=S. Hong;,citation_author=E. Niehaus;,citation_publication_date=2022-05;,citation_cover_date=2022-05;,citation_year=2022;,citation_fulltext_html_url=https://doi.org/10.1016/S2589-7500(22)00048-6;,citation_issue=7;,citation_doi=10.1016/S2589-7500(22)00048-6;,citation_issn=0010-4620;,citation_volume=4;,citation_journal_title=Lancet Digit Health;">
<meta name="citation_reference" content="citation_title=Synergies between centralized and federated approaches to data quality: A report from the national COVID cohort collaborative;,citation_author=Emily R Pfaff;,citation_author=Andrew T Girvin;,citation_author=Davera L Gabriel;,citation_author=Kristin Kostka;,citation_author=Michele Morris;,citation_author=Matvey B Palchuk;,citation_author=Harold P Lehmann;,citation_author=Benjamin Amor;,citation_author=Mark Bissell;,citation_author=Katie R Bradwell;,citation_author=others;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_issue=4;,citation_doi=10.1093/jamia/ocab217;,citation_volume=29;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=Oxford University Press;">
<meta name="citation_reference" content="citation_title=Generalisable long COVID subtypes: Findings from the NIH N3C and RECOVER programmes;,citation_author=Justin T Reese;,citation_author=Hannah Blau;,citation_author=Elena Casiraghi;,citation_author=Timothy Bergquist;,citation_author=Johanna J Loomba;,citation_author=Tiffany J Callahan;,citation_author=Bryan Laraway;,citation_author=Corneliu Antonescu;,citation_author=Ben Coleman;,citation_author=Michael Gargano;,citation_author=others;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_doi=10.1016/j.ebiom.2022.104413;,citation_volume=87;,citation_journal_title=EBioMedicine;,citation_publisher=Elsevier;">
<meta name="citation_reference" content="citation_title=Electronic health records based phenotyping in next-generation clinical trials: A perspective from the NIH health care systems collaboratory;,citation_author=Rachel L Richesson;,citation_author=W Ed Hammond;,citation_author=Meredith Nahm;,citation_author=Douglas Wixted;,citation_author=Gregory E Simon;,citation_author=Jennifer G Robinson;,citation_author=Alan E Bauck;,citation_author=Denise Cifelli;,citation_author=Michelle M Smerek;,citation_author=John Dickerson;,citation_author=others;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_issue=e2;,citation_doi=10.1136/amiajnl-2013-001926;,citation_volume=20;,citation_journal_title=Journal of the American Medical Informatics Association;,citation_publisher=BMJ Publishing Group;">
<meta name="citation_reference" content="citation_title=Outcomes of COVID-19 in patients with cancer: Report from the national COVID cohort collaborative (N3C);,citation_author=Noha Sharafeldin;,citation_author=Benjamin Bates;,citation_author=Qianqian Song;,citation_author=Vithal Madhira;,citation_author=Yao Yan;,citation_author=Sharlene Dong;,citation_author=Eileen Lee;,citation_author=Nathaniel Kuhrt;,citation_author=Yu Raymond Shao;,citation_author=Feifan Liu;,citation_author=Timothy Bergquist;,citation_author=Justin Guinney;,citation_author=Jing Su;,citation_author=Umit Topaloglu;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=20;,citation_doi=10.1200/JCO.21.01074;,citation_volume=39;,citation_journal_title=Journal of Clinical Oncology;">
<meta name="citation_reference" content="citation_title=Association between immune dysfunction and COVID-19 breakthrough infection after SARS-CoV-2 vaccination in the US;,citation_author=Jing Sun;,citation_author=Qulu Zheng;,citation_author=Vithal Madhira;,citation_author=Amy L. Olex;,citation_author=Alfred J. Anzalone;,citation_author=Amanda Vinson;,citation_author=Jasvinder A. Singh;,citation_author=Evan French;,citation_author=Alison G. Abraham;,citation_author=Jomol Mathew;,citation_author=Nasia Safdar;,citation_author=Gaurav Agarwal;,citation_author=Kathryn C. Fitzgerald;,citation_author=Namrata Singh;,citation_author=Umit Topaloglu;,citation_author=Christopher G. Chute;,citation_author=Roslyn B. Mannon;,citation_author=Gregory D. Kirk;,citation_author=Rena C. Patel;,citation_publication_date=2022-02;,citation_cover_date=2022-02;,citation_year=2022;,citation_issue=2;,citation_doi=10.1001/jamainternmed.2021.7024;,citation_issn=2168-6106;,citation_volume=182;,citation_journal_title=Archives of internal medicine (Chicago, Ill. : 1908);,citation_publisher=American Medical Association;">
<meta name="citation_reference" content="citation_title=Synthea™ novel coronavirus (COVID-19) model and synthetic data set;,citation_author=Jason Walonoski;,citation_author=Sybil Klaus;,citation_author=Eldesia Granger;,citation_author=Dylan Hall;,citation_author=Andrew Gregorowicz;,citation_author=George Neyarapally;,citation_author=Abigail Watson;,citation_author=Jeff Eastman;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://www.sciencedirect.com/science/article/pii/S2666521220300077;,citation_doi=doi.org/10.1016/j.ibmed.2020.100007;,citation_issn=2666-5212;,citation_volume=1-2;,citation_journal_title=Intelligence-Based Medicine;">
<meta name="citation_reference" content="citation_title=R for data science;,citation_author=undefined Hadley Wickham;,citation_author=Garrett Grolemund;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://r4ds.hadley.nz/;">
<meta name="citation_reference" content="citation_title=The FAIR guiding principles for scientific data management and stewardship;,citation_author=Mark D Wilkinson;,citation_author=Michel Dumontier;,citation_author=IJsbrand Jan Aalbersberg;,citation_author=Gabrielle Appleton;,citation_author=Myles Axton;,citation_author=Arie Baak;,citation_author=Niklas Blomberg;,citation_author=Jan-Willem Boiten;,citation_author=Luiz Bonino Silva Santos;,citation_author=Philip E Bourne;,citation_author=others;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_doi=10.1038/sdata.2016.18;,citation_volume=3;,citation_journal_title=Scientific data;,citation_publisher=Nature Publishing Group;">
<meta name="citation_reference" content="citation_title=Knitr: A comprehensive tool for reproducible research in R;,citation_author=Yihui Xie;,citation_editor=Victoria Stodden;,citation_editor=Friedrich Leisch;,citation_editor=Roger D. Peng;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_fulltext_html_url=http://www.crcpress.com/product/isbn/9781466561595;,citation_inbook_title=Implementing reproducible computational research;">
<meta name="citation_reference" content="citation_title=Dynamic documents with R and knitr;,citation_author=Yihui Xie;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_fulltext_html_url=http://yihui.org/knitr/;">
<meta name="citation_reference" content="citation_title=Bookdown: Authoring books and technical documents with R markdown;,citation_author=Yihui Xie;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://bookdown.org/yihui/bookdown;">
<meta name="citation_reference" content="citation_title=R markdown: The definitive guide;,citation_author=Yihui Xie;,citation_author=J. J. Allaire;,citation_author=Garrett Grolemund;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://bookdown.org/yihui/rmarkdown;">
<meta name="citation_reference" content="citation_title=R markdown cookbook;,citation_author=Yihui Xie;,citation_author=Christophe Dervieux;,citation_author=Emily Riederer;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://bookdown.org/yihui/rmarkdown-cookbook;">
<meta name="citation_reference" content="citation_title=Associations between HIV infection and clinical spectrum of COVID-19: A population level analysis based on US national COVID cohort collaborative (N3C) data;,citation_author=Xueying Yang;,citation_author=Jing Sun;,citation_author=Rena C Patel;,citation_author=Jiajia Zhang;,citation_author=Siyuan Guo;,citation_author=Qulu Zheng;,citation_author=Amy L Olex;,citation_author=Bankole Olatosi;,citation_author=Sharon B Weissman;,citation_author=Jessica Y Islam;,citation_author=others;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=11;,citation_doi=10.1016/S2352-3018(21)00239-3;,citation_volume=8;,citation_journal_title=The Lancet HIV;,citation_publisher=Elsevier;">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/access.html">Investigating</a></li><li class="breadcrumb-item"><a href="../chapters/understanding.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Understanding the Data</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">The Researcher’s Guide to N3C</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="../The-Researcher-s-Guide-to-N3C.epub" title="Download ePub" class="quarto-navigation-tool px-1" aria-label="Download ePub"><i class="bi bi-journal"></i></a>
    <div class="dropdown">
      <a href="" title="Share" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Share"><i class="bi bi-share"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://twitter.com/intent/tweet?url=|url|">
              <i class="bi bi-bi-twitter pe-1"></i>
            Twitter
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="https://www.facebook.com/sharer/sharer.php?u=|url|">
              <i class="bi bi-bi-facebook pe-1"></i>
            Facebook
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Getting Started</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/story.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">A Research Story</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Data Life Cycle - From Patients to N3C Researchers</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/governance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Governance, Leadership, and Operations Structures</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/onboarding.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Onboarding, Enclave Access, N3C Team Science</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Investigating</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/access.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Getting &amp; Managing Data Access</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/understanding.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Understanding the Data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/tools.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Introducing Enclave Analysis Tools</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/practices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Best Practices for the Research Life Cycle</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/publishing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Publishing and Sharing Your Work</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Special Topics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Help and Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/ml.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Machine Learning</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Back Matter</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/funding.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Funding and Institutional Support</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">References</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#sec-understanding-intro" id="toc-sec-understanding-intro" class="nav-link active" data-scroll-target="#sec-understanding-intro"><span class="header-section-number">7.1</span> Introduction</a></li>
  <li><a href="#sec-understanding-basics" id="toc-sec-understanding-basics" class="nav-link" data-scroll-target="#sec-understanding-basics"><span class="header-section-number">7.2</span> OHDSI Basics</a>
  <ul class="collapse">
  <li><a href="#sec-understanding-basics-ohdsi" id="toc-sec-understanding-basics-ohdsi" class="nav-link" data-scroll-target="#sec-understanding-basics-ohdsi"><span class="header-section-number">7.2.1</span> OHDSI Community</a></li>
  <li><a href="#sec-understanding-basics-omop" id="toc-sec-understanding-basics-omop" class="nav-link" data-scroll-target="#sec-understanding-basics-omop"><span class="header-section-number">7.2.2</span> OMOP Vocabulary</a></li>
  </ul></li>
  <li><a href="#sec-understanding-sets" id="toc-sec-understanding-sets" class="nav-link" data-scroll-target="#sec-understanding-sets"><span class="header-section-number">7.3</span> Concept Sets</a>
  <ul class="collapse">
  <li><a href="#sec-understanding-sets-intro" id="toc-sec-understanding-sets-intro" class="nav-link" data-scroll-target="#sec-understanding-sets-intro"><span class="header-section-number">7.3.1</span> What is a concept set?</a></li>
  <li><a href="#sec-understanding-sets-metadata" id="toc-sec-understanding-sets-metadata" class="nav-link" data-scroll-target="#sec-understanding-sets-metadata"><span class="header-section-number">7.3.2</span> Concept Set Metadata</a></li>
  <li><a href="#sec-understanding-sets-library" id="toc-sec-understanding-sets-library" class="nav-link" data-scroll-target="#sec-understanding-sets-library"><span class="header-section-number">7.3.3</span> N3C Concept Set Library</a></li>
  </ul></li>
  <li><a href="#sec-understanding-ehr" id="toc-sec-understanding-ehr" class="nav-link" data-scroll-target="#sec-understanding-ehr"><span class="header-section-number">7.4</span> EHR-Based Data Beyond OMOP</a>
  <ul class="collapse">
  <li><a href="#sec-understanding-ehr-manifest" id="toc-sec-understanding-ehr-manifest" class="nav-link" data-scroll-target="#sec-understanding-ehr-manifest"><span class="header-section-number">7.4.1</span> Manifest</a></li>
  <li><a href="#sec-understanding-ehr-map" id="toc-sec-understanding-ehr-map" class="nav-link" data-scroll-target="#sec-understanding-ehr-map"><span class="header-section-number">7.4.2</span> Control Map</a></li>
  <li><a href="#sec-understanding-ehr-visit" id="toc-sec-understanding-ehr-visit" class="nav-link" data-scroll-target="#sec-understanding-ehr-visit"><span class="header-section-number">7.4.3</span> Microvisit to Macrovisit Map</a></li>
  <li><a href="#sec-understanding-ehr-harmonized" id="toc-sec-understanding-ehr-harmonized" class="nav-link" data-scroll-target="#sec-understanding-ehr-harmonized"><span class="header-section-number">7.4.4</span> Harmonized values</a></li>
  <li><a href="#sec-understanding-ehr-cohort" id="toc-sec-understanding-ehr-cohort" class="nav-link" data-scroll-target="#sec-understanding-ehr-cohort"><span class="header-section-number">7.4.5</span> Phenotype and cohort</a></li>
  <li><a href="#sec-understanding-ehr-derived" id="toc-sec-understanding-ehr-derived" class="nav-link" data-scroll-target="#sec-understanding-ehr-derived"><span class="header-section-number">7.4.6</span> Derived variables/facts</a></li>
  </ul></li>
  <li><a href="#sec-understanding-pprl" id="toc-sec-understanding-pprl" class="nav-link" data-scroll-target="#sec-understanding-pprl"><span class="header-section-number">7.5</span> Beyond EHR: PPRL-Based Data</a>
  <ul class="collapse">
  <li><a href="#sec-understanding-pprl-intro" id="toc-sec-understanding-pprl-intro" class="nav-link" data-scroll-target="#sec-understanding-pprl-intro"><span class="header-section-number">7.5.1</span> Introduction to PPRL</a></li>
  <li><a href="#sec-understanding-pprl-access" id="toc-sec-understanding-pprl-access" class="nav-link" data-scroll-target="#sec-understanding-pprl-access"><span class="header-section-number">7.5.2</span> PPRL Data Access</a></li>
  <li><a href="#sec-understanding-pprl-mortality" id="toc-sec-understanding-pprl-mortality" class="nav-link" data-scroll-target="#sec-understanding-pprl-mortality"><span class="header-section-number">7.5.3</span> Mortality</a></li>
  <li><a href="#sec-understanding-pprl-variants" id="toc-sec-understanding-pprl-variants" class="nav-link" data-scroll-target="#sec-understanding-pprl-variants"><span class="header-section-number">7.5.4</span> Viral Variant</a></li>
  <li><a href="#sec-understanding-pprl-cms" id="toc-sec-understanding-pprl-cms" class="nav-link" data-scroll-target="#sec-understanding-pprl-cms"><span class="header-section-number">7.5.5</span> Centers for Medicare and Medicaid Services (CMS)</a></li>
  </ul></li>
  <li><a href="#sec-understanding-public" id="toc-sec-understanding-public" class="nav-link" data-scroll-target="#sec-understanding-public"><span class="header-section-number">7.6</span> Public/External Datasets</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/edit/main/chapters/understanding.md" class="toc-action">Edit this page</a></p><p><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/issues/new" class="toc-action">Report an issue</a></p><p><a href="https://github.com/national-covid-cohort-collaborative/guide-to-n3c-v1/blob/main/chapters/understanding.md" class="toc-action">View source</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-understanding" class="quarto-section-identifier"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Understanding the Data</span></span></h1>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliations</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Harold P. Lehmann <a href="https://orcid.org/0000-0002-7698-219X" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://e-catalogue.jhu.edu/medicine/medical-students/subjects-instruction/health-sciences-informatics/">
            Johns Hopkins University
            </a>
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author">Lisa Eskenazi <a href="https://orcid.org/0000-0001-8693-7838" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Johns Hopkins University
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author">Shawn O’Neil <a href="https://orcid.org/0000-0001-6220-7080" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://oneilsh.github.io/">
            University of Colorado Anschutz
            </a>
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author">Thomas Richards <a href="https://orcid.org/0000-0003-4945-9050" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://www.jhu.edu/">
            Johns Hopkins University
            </a>
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author">Kristin Kostka <a href="https://orcid.org/0000-0003-2595-8736" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            The OHDSI Center at the Roux Institute, Northeastern University, Portland, ME &amp; NDORMS, University of Oxford, Oxford, UK
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author">Sigfried Gold <a href="https://orcid.org/0000-0001-7853-6137" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    </div>

<div class="quarto-title-meta">

      
  
    
  </div>
  

</header>

<p><strong>Chapter Leads:</strong> Harold Lehmann, Lisa Eskenazi</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>This chapter is undergoing final edits, including formatting and cross referencing. Edits are expected to complete by June 2023.</p>
</div>
</div>
<section id="sec-understanding-intro" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="sec-understanding-intro"><span class="header-section-number">7.1</span> Introduction</h2>
<p>The purpose of this chapter is to introduce the core building blocks for performing analyses in the Enclave (see <a href="intro.html#sec-intro-enclave">The N3C “Enclave” and Data Access</a> in the Introduction).</p>
<p>We start with the Observational Medical Outcomes Partnership (OMOP) vocabulary, since all data are represented in the OMOP common data model (CDM) format. This chapter will describe the difference between the OMOP concept id unique identifier and the source value contributed by a site, the notion of hierarchy across concepts, and the OHDSI tools outside the Enclave that can help. Because we are working in a secure NCATS Enclave environment, we cannot rely on OHDSI tools outside the Enclave, so knowing the basic applications available within the Enclave and how these data are organized is crucial for the proper conduct of research.</p>
<p>We next present the notion and management of concept sets. These are concept ids that are taken as synonymous, for a given research purpose, and have their own set of tools.</p>
<p>We turn next to constructs—derived variables and facts—derived from concept sets and the raw data, including N3C OMOP additions that build on the OMOP data.</p>
<p>We round out this discussion with an introduction to external datasets that have been brought into the Enclave and the process to bring further datasets in.</p>
<p>Later chapters will give more details on how to use the resources and tools described here.</p>
</section>
<section id="sec-understanding-basics" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="sec-understanding-basics"><span class="header-section-number">7.2</span> OHDSI Basics</h2>
<p>This Observational Health Data Sciences and Informatics (OHDSI) basics section introduces the OHDSI community and provides an introduction to OMOP vocabularies, and vocabulary search tools such as ATHENA and ATLAS. For further OHDSI collaborative knowledge and exploration, refer to <a href="https://ohdsi.github.io/TheBookOfOhdsi/"><em>The Book of OHDSI</em></a> which serves as a central knowledge repository <span class="citation" data-cites="ohdsi_2019">(<a href="../references.html#ref-ohdsi_2019" role="doc-biblioref">OHDSI, 2019</a>)</span>. The Book of OHDSI is a living document, community-maintained through open-source development tools, and evolves continuously. The online version, available for free, always represents the latest version.</p>
<section id="sec-understanding-basics-ohdsi" class="level3" data-number="7.2.1">
<h3 data-number="7.2.1" class="anchored" data-anchor-id="sec-understanding-basics-ohdsi"><span class="header-section-number">7.2.1</span> OHDSI Community</h3>
<p>OHDSI is an international, interdisciplinary research collaborative that promotes open-source research. Its purpose is to create open source solutions that bring out the value of observational health data through large-scale analytics. OHDSI has established an international network of researchers and observational health databases with a central coordinating center housed at Columbia University. More information about OHDSI can be found at <a href="https://www.OHDSI.org" class="uri">https://www.OHDSI.org</a>.</p>
</section>
<section id="sec-understanding-basics-omop" class="level3" data-number="7.2.2">
<h3 data-number="7.2.2" class="anchored" data-anchor-id="sec-understanding-basics-omop"><span class="header-section-number">7.2.2</span> OMOP Vocabulary</h3>
<p>The Book of OHDSI <a href="https://ohdsi.github.io/TheBookOfOhdsi/StandardizedVocabularies.html">states</a>:</p>
<blockquote class="blockquote">
<p>The OMOP Standardized Vocabularies, often referred to simply as “the Vocabulary”, are a foundational part of the OHDSI research network, and an integral part of the CDM. They allow standardization of methods, definitions and results by defining the content of the data, paving the way for true remote (behind the firewall) network research and analytics. Usually, finding and interpreting the content of observational healthcare data, whether it is structured data using coding schemes or laid down in free text, is passed all the way through to the researcher, who is faced with a myriad of different ways to describe clinical events. OHDSI requires harmonization not only to a standardized format, but also to a rigorous standard content.</p>
</blockquote>
<p>N3C relies heavily on the OMOP common data model which is part of the heart of the N3C harmonization process. OMOP, along with other data partners from ACT, TriNetX, and PCORI, ingest their data feeds into a central OMOP N3C repository that is then available within the N3C Enclave for researchers.</p>
<p>The OMOP CDM is an open-source, community standard for observational healthcare data and consists of clinical tables describing conditions and events from patient records and vocabulary tables describing clinical concepts. The central table in the OMOP vocabulary system is the table, <code>concept</code>.</p>
<div id="fig-understanding-010-concept" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-010-concept.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Sample rows from the 6.9-million row table of concepts imported from OMOP"><img src="images/understanding/fig-understanding-010-concept.png" class="img-fluid figure-img" alt="Sample rows from the 6.9-million row table of concepts imported from OMOP"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.1: Sample rows from the 6.9-million row table of concepts imported from OMOP</figcaption>
</figure>
</div>
<section id="sec-understanding-basics-omop-vocab" class="level4" data-number="7.2.2.1">
<h4 data-number="7.2.2.1" class="anchored" data-anchor-id="sec-understanding-basics-omop-vocab"><span class="header-section-number">7.2.2.1</span> Vocabulary Concepts</h4>
<p>The meaning of every data element in the OMOP vocabulary, and many of the values of the data themselves, are represented by codes in the source data that get mapped to OMOP concept ids. Suppose a physician diagnoses a patient with diabetes and types “I10” into the EHR (which uses the ICD10 vocabulary). The “I10” value is stored in OMOP as the <code>source_value</code>, and is then translated via the OMOP Vocabulary to the standard concept id, “320128”.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> N3C analyses use <em>standard</em> concept ids, as opposed to specific codes that aren’t necessarily consistent across hospitals.</p>
<p><em>The Book of OHDSI</em> has a detailed chapter on <a href="https://ohdsi.github.io/TheBookOfOhdsi/StandardizedVocabularies.html">Standardized Vocabularies</a>. The OHDSI vocabularies are in a standardized CDM structure that houses existing vocabularies used in the public domain. This CDM compiles standards from disparate public and private sources as well as some OMOP grown concepts.</p>
<p>Concepts are related to each other. For instance, hypertrophic cardiomyopathy (concept id <code>4124693</code>) is a type of cardiomyopathy (concept id <code>321319</code>). The <a href="https://ohdsi.github.io/CommonDataModel/cdm60.html#CONCEPT_RELATIONSHIP"><code>concept_relationship</code></a> table stores these relationships. While analysts are welcome to use this table in their queries, there are other approaches to doing the same. (See <a href="../chapters/understanding.html#sec-understanding-sets">concept sets</a>.)</p>
<div id="fig-understanding-020-concept-relationship-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-020-concept-relationship-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="A sample from the 54.4 million-row table imported from OMOP, focused on the cardiomyopathy concept id, 321319"><img src="images/understanding/fig-understanding-020-concept-relationship-1.png" class="img-fluid figure-img" alt="A sample from the 54.4 million-row table imported from OMOP, focused on the cardiomyopathy concept id, 321319"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.2: A sample from the 54.4 million-row table imported from OMOP, focused on the cardiomyopathy concept id, 321319</figcaption>
</figure>
</div>
<p>Every relationship type connecting two OMOP concepts has a converse. In the <code>concept_relationship</code> table, this means a parent-child relationship will be represented (redundantly) by two records. Hypertrophic cardiomyopathy (concept id 4124693) is a type of cardiomyopathy (concept id 321319):</p>
<div id="fig-understanding-030-concept-relationship-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-030-concept-relationship-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="The converse representation of @fig-understanding-020-concept-relationship-1 row 47"><img src="images/understanding/fig-understanding-030-concept-relationship-2.png" class="img-fluid figure-img" alt="The converse representation"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.3: The converse representation of <a href="#fig-understanding-020-concept-relationship-1">Figure&nbsp;<span>7.2</span></a> row 47</figcaption>
</figure>
</div>
<p>The hierarchical relationship between the two concepts is represented once (above) with a “Subsumes” record and again (below) with an “Is a” record. <code>concept_relationship</code> contains only direct relationships between pairs of concepts. For hierarchical relationships, more distant relationships are found in the <a href="https://ohdsi.github.io/CommonDataModel/cdm60.html#CONCEPT_ANCESTOR"><code>concept_ancestor</code></a> table.</p>
<p>The <code>concept_relationship</code> table contains many other relationship types beyond hierarchical relationships: “Mapped from”, “Maps to”, “Has brand name of”, “Consists of”, etc.</p>
<p><a href="https://athena.ohdsi.org/search-terms/terms?query">ATHENA</a> is an OHDSI tool for exploring OMOP vocabularies and even downloading them; its primary use in N3C is for looking up individual terms. More details on using ATHENA to search terms are documented on OHDSI’s publicly-viewable GitHub <a href="https://github.com/OHDSI/Athena">repository</a>. <a href="https://atlas.ohdsi.org/#/home">ATLAS</a> is an OHDSI tool to explore concepts and their relationships. See section below on <a href="../chapters/understanding.html#sec-understanding-sets">concept sets</a> that describes further uses of the ATLAS tool.</p>
<p>The OMOP vocabulary is updated regularly to keep up with the continual evolution of sources. Its vocabulary maintenance and improvement is an ongoing activity that requires community participation and support. Releases and release notes are published via Git Releases on the OMOP Vocabulary GitHub repository. The most recent specification documentation can be found at <a href="https://ohdsi.github.io/CommonDataModel/cdm53.html">OMOP Common Data Model</a>. New versions of the OMOP vocabulary are downloaded monthly by the Enclave managers.</p>
<p>The OMOP CDM assembles these relationships in the <a href="https://ohdsi.github.io/CommonDataModel/cdm60.html#Vocabulary_Tables">Standardized vocabulary</a> section of the CDM. These tables are available to N3C researchers, along with the <a href="https://ohdsi.github.io/CommonDataModel/cdm60.html#Clinical_Data_Tables">Standardized clinical data</a> tables that store the patients’ data (expressed using the concept ids). Data are available as well in the <a href="https://ohdsi.github.io/CommonDataModel/cdm60.html#Health_System_Data_Tables">Standardized health system</a>, <code>Payer_plan_period</code> (not <code>cost</code>), and <a href="https://ohdsi.github.io/CommonDataModel/cdm60.html#Standardized_Derived_Elements">Standardized derived elements</a> tables.</p>
<div id="fig-understanding-040-omop" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-040-omop.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="OMOP Data model"><img src="images/understanding/fig-understanding-040-omop.png" class="img-fluid figure-img" alt="omop"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.4: OMOP <a href="https://ohdsi.github.io/CommonDataModel">Data model</a></figcaption>
</figure>
</div>
<p>To learn more about OMOP and get acquainted with the CDM used in the N3C Data Enclave and its functionality, see <a href="https://unite.nih.gov/workspace/report/ri.report.main.report.1f3c0dae-0191-4f52-b69e-4e0587e24b3b">OMOP 101: A Crash Course in OMOP Standard Vocabulary</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> and <a href="support.html"><span>Chapter&nbsp;11</span></a> in this guide.</p>
<p>The figure below shows all OMOP tables within the N3C Enclave.</p>
<div id="fig-understanding-050-lds" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-050-lds.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Note that this folder contains tables (e.g., conditions-to-microvisits) that are unique to N3C. The SafeHarbor release has a comparable folder"><img src="images/understanding/fig-understanding-050-lds.png" class="img-fluid figure-img" alt="n3c tables"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.5: Note that this folder contains tables (e.g., conditions-to-microvisits) that are unique to N3C. The SafeHarbor release has a comparable folder</figcaption>
</figure>
</div>
</section>
<section id="sec-understanding-basics-omop-domains" class="level4" data-number="7.2.2.2">
<h4 data-number="7.2.2.2" class="anchored" data-anchor-id="sec-understanding-basics-omop-domains"><span class="header-section-number">7.2.2.2</span> Domains</h4>
<p>Tables may contain data from more than one “domain”. There are currently 36 Domains in OMOP; the 5 primary clinical Domains are Drug, Condition, Observation, Procedure, and Measurement, plus others. They classify concepts in the concept table.</p>
<div id="fig-understanding-060-domains" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-060-domains.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Listed in decreasing order by the number of concepts for which we have data in N3C"><img src="images/understanding/fig-understanding-060-domains.png" class="img-fluid figure-img" alt="Listed in decreasing order by the number of concepts for which we have data in N3C"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.6: Listed in decreasing order by the number of concepts for which we have data in N3C</figcaption>
</figure>
</div>
</section>
<section id="sec-understanding-basics-omop-updates" class="level4" data-number="7.2.2.3">
<h4 data-number="7.2.2.3" class="anchored" data-anchor-id="sec-understanding-basics-omop-updates"><span class="header-section-number">7.2.2.3</span> Vocabulary Updates</h4>
<p>As stated above, source vocabularies change. For instance, at the start of the pandemic, there were no ICD10 codes for “COVID-19 infection”, and therefore, there were no OMOP vocabulary codes for it and, therefore, no N3C codes. When we need a code, but there is none in the OMOP vocabulary, we create a <em>local code</em>. When the OMOP vocabulary team releases a newer, relevant code to replace the local code, the Enclave attempts to contact everyone who has employed the local code, notifying them of the change.</p>
<p>More subtle issues are when the source vocabulary managers change or decommission existing codes or add new codes (that do not have a local-code equivalent), and the OMOP managers incorporate those changes in the vocabularies brought into the Enclave every month. Another subtlety is when the relationships between codes are changed. The Enclave attempts to contact those who have used codes related to the changed code. Generally, the analyst has used those codes in concept sets, to which topic we now turn.</p>
</section>
</section>
</section>
<section id="sec-understanding-sets" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="sec-understanding-sets"><span class="header-section-number">7.3</span> Concept Sets</h2>
<p>Any study or analysis performed in the Enclave will start with identifying meaningful clinical conditions and events in patient data. We attempt to determine the presence or absence of clinical phenomena in patient history through the presence or absence of certain concept codes in patient records <span class="citation" data-cites="gold_2021 gold_2018">(<a href="../references.html#ref-gold_2018" role="doc-biblioref">Gold et al., 2018</a>, <a href="../references.html#ref-gold_2021" role="doc-biblioref">2021</a>)</span>. For any given phenomenon used in an analysis, however, a single concept code will seldom be sufficient, and concept <em>sets</em> are used.</p>
<p>There are many reasons to specify a concept set:</p>
<ul>
<li>Variables in your study are implemented using concept sets.</li>
<li>Having dozens or hundreds of concept ids in a single concept set makes analysis much easier.</li>
<li>It makes the analysis easier to peer review.</li>
<li>It makes your research coherent with the research of others who use the same concept set(s).</li>
<li>Chunking concept-ids together into concept sets with different names and intentions makes your own (research) intentions clearer.</li>
</ul>
<p><em>If you are publishing your results, concept sets are considered a publicly viewable part of your work.</em> Such public viewing is important.</p>
<ul>
<li>It is required by the Strengthening the <em>Reporting of Observational Studies in Epidemiology</em> <a href="https://www.equator-network.org/reporting-guidelines/strobe/">STROBE</a> guidelines for publishing observational research.</li>
<li>Open Science expects reproducibility, and reproducibility in EHR-based research requires knowing how variables (concept sets) were defined.</li>
<li>It enables other N3C researchers to use your concept sets, which is what makes research coherent across N3C projects.</li>
</ul>
<p>Concept sets that are “finalized” within N3C will have information about them posted to our publicly-viewable GitHub <a href="https://github.com/National-COVID-Cohort-Collaborative/Data-Ingestion-and-Harmonization/tree/master/concept_sets">repository</a>. Such posting is part of the process known as making research <a href="https://www.force11.org/group/fairgroup/fairprinciples">FAIR</a> (findable, accessible, interoperable, reusable). This repository is functionally our <em>Shared Variable Library</em>. Other variable definitions beyond concept sets are posted there (formulas, harmonized values, cohort definitions).</p>
<p>The concept set content that will be made public in <a href="https://zenodo.org/">Zenodo</a> and/or GitHub are:</p>
<ul>
<li>Concept set name, version number, date of finalization</li>
<li>Intention</li>
<li>Limitations (of the concept-ids within the concept set)</li>
<li>Issues (with respect to data in the N3C Data Enclave, so analysts are forewarned)</li>
<li>Provenance (how the concept set was assembled, which communicates how trustworthy the concept set is)</li>
<li>Attributions (who did what)</li>
<li>The list of codes (in JavaScript Object Notation (JSON) format, which can be imported directly into Observational Health Data Sciences &amp; Informatics (OHDSI)’s Atlas tool)</li>
<li>Zenodo persistent document object identifier (DOI)</li>
</ul>
<section id="sec-understanding-sets-intro" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="sec-understanding-sets-intro"><span class="header-section-number">7.3.1</span> What is a concept set?</h3>
<p><em>The Book of OHDSI</em> explains <a href="https://ohdsi.github.io/TheBookOfOhdsi/Cohorts.html#conceptSets">concept sets</a> as:</p>
<blockquote class="blockquote">
<p>A concept set is an expression representing a list of concepts that can be used as a reusable component in various analyses. It can be thought of as a standardized, computer-executable equivalent of the code lists often used in observational studies. A concept set expression consists of a list of concepts with the following attributes:</p>
</blockquote>
<ul>
<li><strong>Exclude</strong>: Exclude this concept (and any of its descendants if selected) from the concept set.</li>
<li><strong>Descendants</strong>: Consider not only this concept, but also all of its descendants.</li>
<li><strong>Mapped</strong>: Allow to search for non-standard concepts.</li>
</ul>
<p>While ATLAS concept sets may contain standard and non-standard concepts, Enclave concept sets may include “standard” (SNOMED-based) concepts only. Thus, the “Mapped” attribute is rarely helpful, as it applies only if using source data.</p>
<p>For example, a concept set expression could contain two concepts as depicted in <em>The Book of OHDSI’s</em> <a href="https://ohdsi.github.io/TheBookOfOhdsi/Cohorts.html#tab:conceptSetExpression">Cohort chapter</a>. Here we include concept <a href="https://athena.ohdsi.org/search-terms/terms/4329847">4329847</a> (“Myocardial infarction”) and all of its descendants, but exclude concept <a href="https://athena.ohdsi.org/search-terms/terms/314666">314666</a> (“Old myocardial infarction”) and all of its descendants.</p>
<div id="tbl-understanding-concept-set" class="anchored">
<table class="table">
<caption>Table&nbsp;7.1: Representation, from Book of OHDSI, of the Myocardial Infarction subtree of standard concepts</caption>
<colgroup>
<col style="width: 17%">
<col style="width: 38%">
<col style="width: 14%">
<col style="width: 18%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th>Concept Id</th>
<th>Concept Name</th>
<th>Excluded</th>
<th>Descendants</th>
<th>Mapped</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>4329847</td>
<td>Myocardial infarction</td>
<td>NO</td>
<td>YES</td>
<td>NO</td>
</tr>
<tr class="even">
<td>314666</td>
<td>Old myocardial infarction</td>
<td>YES</td>
<td>YES</td>
<td>NO</td>
</tr>
</tbody>
</table>
</div>
<p>As shown in <a href="#tbl-understanding-concept-set">Table&nbsp;<span>7.1</span></a>, this will include “Myocardial infarction” and all of its descendants except “Old myocardial infarction” and its descendants. In total, this concept set expression implies nearly a hundred Standard Concepts. These Standard Concepts in turn reflect hundreds of source codes (e.g.&nbsp;ICD-9 and ICD-10 codes) that may appear in the various databases.</p>
<div id="fig-understanding-080-mi-hierarchy" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-080-mi-hierarchy.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Representation, from Book of OHDSI, of the Myocardial Infarction subtree of standard concepts"><img src="images/understanding/fig-understanding-080-mi-hierarchy.png" class="img-fluid figure-img" alt=""></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.7: Representation, from Book of OHDSI, of the Myocardial Infarction subtree of standard concepts</figcaption>
</figure>
</div>
<p>ATLAS is used by the OHDSI community to construct concept sets. Best practice is for each research institution to have its own CDM instance. The Enclave, for technical reasons, cannot implement ATLAS. In the future, N3C hopes to provide a public tool but it is unclear when this release will occur. Users without their own local ATLAS may explore its utility using a publicly available one on OHDSI.org. <em>Do beware</em> all public tools are intended to be a demonstration site to showcase the power of the open-source tool stack. They are not intended for active research projects. These deployments are not running on a real OMOP CDM. They use synthetic data like the CMS Synthetic Public Use Files (SYNPUF) in the OMOP CDM format. The vocabulary schema will reflect whatever version of the vocabulary was in place when that OMOP CDM ETL was run. It will not reflect the most recent version of the OMOP vocabulary, which may be problematic for research projects that use newly approved vaccines or medicines. You should not make the habit of saving content within the public instance. It can be reset and wiped at any point in time. If you enjoy using ATLAS, consider spinning up your own instance via Broadsea, OHDSI in a box, or other OHDSI Community resources to jump start users with their own ATLAS deployment. It is very easy to do and runs effectively, even on synthetic data. Resources are available via the OHDSI ATLAS workgroup. While it is a handy complement to the Enclave, do focus on the N3C products: The N3C concept set library and the concept set editor. The Enclave builds its own OMOP vocabulary releases from downloading the vocabulary tables from ATHENA.</p>
<p>Because of these limitations, also <em>do beware</em> that ATLAS “record counts” (the number of patients with data expressing the given concept id) are <em>not</em> linked to N3C and do not represent the counts or distribution of concept ids within the Enclave.</p>
</section>
<section id="sec-understanding-sets-metadata" class="level3" data-number="7.3.2">
<h3 data-number="7.3.2" class="anchored" data-anchor-id="sec-understanding-sets-metadata"><span class="header-section-number">7.3.2</span> Concept Set Metadata</h3>
<p>The work on concept sets does not take place in a vacuum. It is assumed that you are part of a project team, which has the following Subject Matter Expertise (SME): Research/Clinical, Vocabulary, Informatics, and Statistical/Analytic. Each of these SMEs will be called upon in your work.</p>
<p>Intention, Limitations, and Provenance comprise concept set metadata.</p>
<section id="sec-understanding-sets-metadata-intention" class="level4" data-number="7.3.2.1">
<h4 data-number="7.3.2.1" class="anchored" data-anchor-id="sec-understanding-sets-metadata-intention"><span class="header-section-number">7.3.2.1</span> Intention</h4>
<p>Intention communicates in sentences more than the name can. For instance, is the concept set intended to be “broad”, and sensitive, to capture as many cases as possible, leaving to downstream analysts to winnow the set of patients down? Or is it “narrow”, and specific? Is it meant to be definitional (these codes “mean” diabetes) or indicative (the codes tell me that you have diabetes, e.g., “retinopathy due to diabetes mellitus”; the codes suggest chronic lung disease, e.g., “infertility due to cystic fibrosis”). “Intention” can also indicate whether the concept set has clinical manifestations or not (e.g., “Sickle Cell” indicates a genetic condition, but unclear if it’s homozygous or heterozygous, the former being clinical and the latter often not). It is true that a later analyst will look through the list of codes, but having an explicit intention helps that analyst screen potential concept sets.</p>
</section>
<section id="sec-understanding-sets-limitations" class="level4" data-number="7.3.2.2">
<h4 data-number="7.3.2.2" class="anchored" data-anchor-id="sec-understanding-sets-limitations"><span class="header-section-number">7.3.2.2</span> Limitations</h4>
<p>Limitations communicates edge cases and caveats to the analyst. “Issues” communicates performance with the Enclave data. This performance could include the number of codes contributing the majority of the data (e.g., from Term Usage) or the distribution of values across sites, in the case of lab tests.</p>
</section>
<section id="sec-understanding-sets-metadata-provenance" class="level4" data-number="7.3.2.3">
<h4 data-number="7.3.2.3" class="anchored" data-anchor-id="sec-understanding-sets-metadata-provenance"><span class="header-section-number">7.3.2.3</span> Provenance</h4>
<p>Provenance communicates how the concept set was assembled. It should include any authorities consulted (the literature, the Value Set Authority Center, the domain team members, etc.) and a sense of what modifications, if any, were made from that authoritative beginning. (Of note, “ATLAS” is not an authoritative beginning, unless there is annotation on the ATLAS site of the authoritative source for that specific concept set.)</p>
</section>
</section>
<section id="sec-understanding-sets-library" class="level3" data-number="7.3.3">
<h3 data-number="7.3.3" class="anchored" data-anchor-id="sec-understanding-sets-library"><span class="header-section-number">7.3.3</span> N3C Concept Set Library</h3>
<p>The concept sets already created in the Enclave can be browsed in the <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.5a6c64c0-e82b-4cf8-ba5b-645cd77a1dbf">concept set browser</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>, illustrated in <a href="#fig-understanding-090-concept-set-versions">Figure&nbsp;<span>7.8</span></a>. This is the first step for deciding whether a new concept is needed or if an existing concept set can be chosen for inclusion into an analysis as a building block for an analytic variable or cohort definition. We recommend that you use or build from existing concept sets, especially the N3C Recommended concepts sets, if they serve your research question. A workflow for performing this series of tasks is shown in <a href="#fig-understanding-090-concept-set-versions">Figure&nbsp;<span>7.8</span></a>.</p>
<div id="fig-understanding-090-concept-set-versions" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-090-concept-set-versions.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="An illustration of the Myocardial Infarction concept set and container and versions in the concept set browser"><img src="images/understanding/fig-understanding-090-concept-set-versions.png" class="img-fluid figure-img" alt="An illustration of the Myocardial Infarction concept set and container and versions in the concept set browser"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.8: An illustration of the Myocardial Infarction concept set and container and versions in the concept set browser</figcaption>
</figure>
</div>
<p>When created from scratch, a <em>container</em> is created, so multiple <em>versions</em> of the concept sets can be supported. New versions are created as the team gets better facility with the tool, as the team understands better from domain experts subtleties in the intention of the concept set, or as changes are made to the vocabularies, by OHDSI or the source vocabulary stewards. Containers are indicated by blue icons; versions, by brown icons, as illustrated in <a href="#fig-understanding-090-concept-set-versions">Figure&nbsp;<span>7.8</span></a>.</p>
<div id="fig-understanding-100-concept-set-flowsheet" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><a href="images/understanding/fig-understanding-100-concept-set-flowsheet.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Workflow for finding, extending, and creating concept sets"><img src="images/understanding/fig-understanding-100-concept-set-flowsheet.png" class="img-fluid figure-img" alt="Workflow for finding, extending, and creating concept sets"></a></p>
<figcaption class="figure-caption">Figure&nbsp;7.9: Workflow for finding, extending, and creating concept sets</figcaption>
</figure>
</div>
<p>*New container name should be similar to source container, but different enough to communicate altered intention. MS=Manuscript</p>
<p>The browse screen allows you to filter to concept sets relevant to you. Once you have identified a concept set that you might use, drilling into it provides data about that concept set: versions, metadata, concept expressions, concepts. In reviewing the concept set (generally by drilling to the most recent version), it may be worth exploring similar concept sets using the concept set Overlap tab. At this point, you may be satisfied that an existing concept set serves your needs. If not, you can create a new one — either an entirely new concept set, or a new version of an existing concept set, modifying it to meet your requirements. If you are creating a new version within an existing container, please contact the concept set owner to ensure that your update is in the right spirit. If not, clone the existing version into a new container, naming the container with a name that connotes the difference between the existing and the new concept sets.</p>
<p>If you use a concept set in a published manuscript, make it publicly available. See the next section, <a href="@sec-understanding-sets-library-published">Published Concept Sets</a>.</p>
<p>Additional documentation on the concept set browser and editor can be found on the <a href="https://unite.nih.gov/workspace/documentation/product/concept-set-builder">Enclave</a>. A <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.e7b83a8c-545e-49ac-8714-f34bfa7f7767?view=focus&amp;Id=7">tutorial</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> is available that explores these tools further.</p>
<section id="sec-understanding-sets-library-published" class="level4" data-number="7.3.3.1">
<h4 data-number="7.3.3.1" class="anchored" data-anchor-id="sec-understanding-sets-library-published"><span class="header-section-number">7.3.3.1</span> Published Concept Sets</h4>
<p>“Publication” includes 3 ways of making a concept set available to other researchers. Provisionally Approved and N3C Recommended are automatically submitted to Zenodo, to generate a permanent DOI URL that can be used in publication or in sharing concept set contents with analysts outside the Enclave. Any study manuscript for publication should ensure that any concept set used is either Provisionally Approved or N3C Recommended. The Zenodo URL should appear in the resulting publication.</p>
<section id="n3c-recommended-concept-sets---pre-defined-concept-set-commonly-used" class="level5" data-number="7.3.3.1.1">
<h5 data-number="7.3.3.1.1" class="anchored" data-anchor-id="n3c-recommended-concept-sets---pre-defined-concept-set-commonly-used"><span class="header-section-number">7.3.3.1.1</span> N3C Recommended concept sets - pre-defined concept set commonly used</h5>
<p>The <a href="https://covid.cd2h.org/liaisons">N3C Data Liaisons and Logic Liaisons</a>, in partnership with the Domain Teams, have selected common medical features often required for broad use in research that support the analytics pipelines. The original N3C Recommended concept sets were selected based on</p>
<ol type="1">
<li><p>the comorbid conditions identified by the <a href="https://www.cdc.gov/coronavirus/2019-ncov/need-extra-precautions/people-with-medical-conditions.html">US Centers for Disease Control (CDC) definitions for COVID-19</a> as increasing risk of severe COVID-19</p></li>
<li><p>the comorbid conditions identified in the <a href="https://healthcaredelivery.cancer.gov/seermedicare/considerations/comorbidity.html#:~:text=The%20Charlson%20Comorbidity%20Index%20was,were%20included%20in%20the%20index.">Charlson Comorbidity Index</a>, and</p></li>
<li><p>others as requested by N3C leadership for use in the Logic Liaison templates.</p></li>
</ol>
<p>The N3C Recommended concept sets have been vetted by clinical experts (typically N3C Domain Team leads) and the Data and Logic Liaison teams according to the below refinement and finalization process. <em>The latest version of the listed concept sets should always be used</em>, as earlier versions represent iterations and have not passed the final review.</p>
<section id="refinement-process" class="level6" data-number="7.3.3.1.1.1">
<h6 data-number="7.3.3.1.1.1" class="anchored" data-anchor-id="refinement-process"><span class="header-section-number">7.3.3.1.1.1</span> Refinement Process</h6>
<p>The clinical experts and the N3C Data Liaisons and Logic Liaisons perform the following steps in establishing an intentional N3C Recommended concept set:</p>
<ul>
<li>Select relevant parent concepts and their descendants from OMOP standard codes, using Charlson Comorbidity description or the relevant section the referenced CDC page to establish intention.</li>
<li>Utilize the OHDSI Atlas tool to explore the OMOP hierarchy to look for other potential parent codes and also to remove child codes that are non-specific to the intended scope of the concept set.</li>
<li>Compare the draft concept set to other related concept sets, using the Concept Set Overlap feature of the Concept Set Editor. This facilitates review of any codes that do not overlap with those in value sets from reputable stewards such as the National Library of Medicine’s Value Set Authority Center (VSAC) and the Healthcare Cost and Utilization Project (HCUP) and from existing similar concept sets within the Concept Set Browser. Add and remove concepts (and their descendants) as per recommendations from the clinical experts.</li>
<li>Reduce the intentional concept set expression as parsimonious as possible, retaining all the approved concepts collected in prior steps.</li>
<li>Iterate as needed.</li>
<li>Document metadata (properties) in the Concept Set Editor for each created concept set to include: Intention, Limitations, Provenance, and at least one vocabulary and one clinical review.</li>
<li>Present for final vetting at the Data Liaison meeting for N3C Recommended designation.</li>
</ul>
<p>These concept sets are in queue to be published as pdf (Properties) and json (concept ids) in Zenodo, to be available to researchers outside the Enclave. Once published, their Zenodo DOI will be posted to the Zenodo Property of the version that was published. The N3C Recommended concept sets are currently published to the <a href="https://github.com/National-COVID-Cohort-Collaborative/Data-Ingestion-and-Harmonization/tree/master/concept_sets/n3c-recommended-csets-json">N3C GitHub</a> and are available to researchers outside the Enclave.</p>
<p>The N3C Recommended concept sets listed below are used as the default concept sets in the Logic Liaison templates <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-BE5C652&amp;view=focus">COVID-19 Diagnosed or Lab Confirmed Patients</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> and <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-BA3B835&amp;view=focus">All Patients</a> which then populate the <a href="https://unite.nih.gov/workspace/slate/documents/phenotype-dashboard">N3C Phenotype Explorer</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> and <a href="https://covid.cd2h.org/dashboard/">N3C Public Health Dashboards</a>.</p>
</section>
</section>
<section id="provisionally-approved" class="level5" data-number="7.3.3.1.2">
<h5 data-number="7.3.3.1.2" class="anchored" data-anchor-id="provisionally-approved"><span class="header-section-number">7.3.3.1.2</span> Provisionally Approved</h5>
<p>The third type of publication is Provisionally Approved. This status is automatically established once a concept set achieves at least one positive Vocabulary Review and one positive Clinical Review.</p>
</section>
</section>
<section id="sec-understanding-sets-library-review" class="level4" data-number="7.3.3.2">
<h4 data-number="7.3.3.2" class="anchored" data-anchor-id="sec-understanding-sets-library-review"><span class="header-section-number">7.3.3.2</span> Concept set reviews and validation</h4>
<p>Why should anyone trust a concept set? This trust is needed both within the project team and outside the team. The project team needs to trust a concept set in an analysis; someone outside the team needs to assess its trust in deciding whether to (re) use it in their own analysis.</p>
<p>“Trust” in a concept set derives from a number of factors. First, do we trust the people who assembled the concept set and its process? Second, do we believe that the concept set encodes the concepts we want it to encode, and excludes those we do not? Third, does it perform within the Enclave data the way we expect it to?</p>
<p>Re “the people” and “the process”: The concept set meta-data, especially the identity of the author and reviewers, and the Provenance, provide the necessary information.</p>
<p>Another means of deciding on a level of trust is the Reviews. The Enclave supports two types of reviews.</p>
<ul>
<li>“Vocabulary review”: Here, a terminologist or informatician familiar with the OMOP vocabulary has reviewed the process documented (in Provenance) or in real time (as part of work with the Domain Team) and documents satisfaction with what codes are included and what are excluded.</li>
<li>“Clinical review”: Here, a clinical member of the Domain Team or project has reviewed the codes to confirm that they match the “Intention” of the concept set.</li>
</ul>
<p>A third means of deciding on a level of trust is looking to see what research projects have used the concept set. The more authoritative (and published) those projects are, the more authority they have. Of course, in some cases, a concept set “going viral” may simply be a popularity contest and not an indicator of correctness (false positives). At the same time, linking of a concept set to a research project is a manual process, which is often not done (false negatives).</p>
<p>Note that this entire scheme of authority and validity depends on adequate documentation of properties. There are many enticing concept sets that do not have properties filled, reviews performed, nor projects linked. Please ensure that your concept sets are not missing these key components.</p>
<p>Note, as well, that, if you publish a paper, using a concept set, that concept set should be published as well. The Properties and Reviews will be published, whether missing or not.</p>
</section>
</section>
</section>
<section id="sec-understanding-ehr" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="sec-understanding-ehr"><span class="header-section-number">7.4</span> EHR-Based Data Beyond OMOP</h2>
<p>Data tables of use or interest to analysts found in either the LDS or Safe Harbor Release are the following:</p>
<section id="sec-understanding-ehr-manifest" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="sec-understanding-ehr-manifest"><span class="header-section-number">7.4.1</span> Manifest</h3>
<p>This table, found in the folder along with the OMOP tables, provides data about each data partner that analysts should find helpful when trying to understand variation in the data: CDM type (ACT, OMOP, PCORNet, TriNetX) and version, whether dates are shifted (and how much), and date of last contribution.</p>
</section>
<section id="sec-understanding-ehr-map" class="level3" data-number="7.4.2">
<h3 data-number="7.4.2" class="anchored" data-anchor-id="sec-understanding-ehr-map"><span class="header-section-number">7.4.2</span> Control Map</h3>
<p>For every “case” generated locally by the Phenotype algorithm, the algorithm also identifies 2 “controls”, that are matched by age, sex, race, and ethnicities, including matched by missing values. While it may be tempting to use this map to identify controls for your own study, too many patients who were initial “controls” may have become “cases”, or their true status is unknown, because of home or other non-site testing. Other matching methods (e.g., propensity scores) are more trustworthy.</p>
</section>
<section id="sec-understanding-ehr-visit" class="level3" data-number="7.4.3">
<h3 data-number="7.4.3" class="anchored" data-anchor-id="sec-understanding-ehr-visit"><span class="header-section-number">7.4.3</span> Microvisit to Macrovisit Map</h3>
<p>While admissions to the hospital are recorded in the Visit_Occurrence table, the end-date is not always so recorded. Now, in many hospitals, procedures performed during a hospitalization may be recorded in the EHR as an “encounter”. So an admission may be represented in the visit_occurrence table as a string of such “encounters”. We define a macrovisit as a merge of chronological, overlapping inpatient and other longitudinal facility visits, to which we add any other types of visits (outpatient, telehealth, etc) that occur during the merged interval. See <a href="https://unite.nih.gov/workspace/report/ri.report.main.report.c9e2ca50-860c-4988-93c6-f5b1d9d915ed">Enclave</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>.</p>
</section>
<section id="sec-understanding-ehr-harmonized" class="level3" data-number="7.4.4">
<h3 data-number="7.4.4" class="anchored" data-anchor-id="sec-understanding-ehr-harmonized"><span class="header-section-number">7.4.4</span> Harmonized values</h3>
<p>It is no surprise that, across over 70 data partners, and the many included sites, different institutions measure the same analyte in different ways and report the results with different units of measure. N3C has spent a fair amount of effort to harmonize units (and therefore values; think degrees Fahrenheit and degrees Celsius), so analysts have a level playing field for their analyses. In addition, many units of measure are missing, so, as part of that effort, an algorithm has been developed to use the pooled data across the Enclave to help us infer units (and therefore, values). This strategy has led to the “rescue” of over 78% of records where units of measure were missing; see <span class="citation" data-cites="bradwell_2022">Bradwell et al. (<a href="../references.html#ref-bradwell_2022" role="doc-biblioref">2022</a>)</span>. This harmonization was done for about 70 measurements so far. The results are found in the Measurement table, as <code>harmonized_value_as_number</code>, <code>harmonized_unit_concept_id</code> and <code>unit_concept_id_or_inferred_unit_concept_id</code>.</p>
</section>
<section id="sec-understanding-ehr-cohort" class="level3" data-number="7.4.5">
<h3 data-number="7.4.5" class="anchored" data-anchor-id="sec-understanding-ehr-cohort"><span class="header-section-number">7.4.5</span> Phenotype and cohort</h3>
<p>A <em>phenotype</em> is a grouping of related terms, or observable characteristics that could be applied to a person, disease trait, medical condition or events <span class="citation" data-cites="richesson_2013">(<a href="../references.html#ref-richesson_2013" role="doc-biblioref">Richesson et al., 2013</a>)</span>. Examples would be, “has Acute COVID”; “treated with ampicillin”.<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a></p>
<p>A <em>cohort</em> is a set of persons who satisfy one or more inclusion criteria for a duration of time. Data ingestion is the process of importing data meeting a phenotype definition into a database which can then be used for research. In this section, we will discuss existing tools and methodologies for identifying codes and terms, creating phenotypes, managing vocabulary concept sets, and how to ingest codes into the N3C Enclave database that will enable research using the N3C basic and supplemental datasets.</p>
</section>
<section id="sec-understanding-ehr-derived" class="level3" data-number="7.4.6">
<h3 data-number="7.4.6" class="anchored" data-anchor-id="sec-understanding-ehr-derived"><span class="header-section-number">7.4.6</span> Derived variables/facts</h3>
<p>Facts are designations that a patient falls into a category (at a point in time). Thus, <code>confirmed_covid_patient</code> or <code>TOBACCOSMOKER_indicator</code> is set as “1” (true) if certain conditions are met in the template <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-BA3B835&amp;view=focus">Logic Liaison Template All Patients Facts</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>. Other Logic Liaison Templates include many derived facts of value to analysis. See <a href="tools.html"><span>Chapter&nbsp;8</span></a>.</p>
</section>
</section>
<section id="sec-understanding-pprl" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="sec-understanding-pprl"><span class="header-section-number">7.5</span> Beyond EHR: PPRL-Based Data</h2>
<section id="sec-understanding-pprl-intro" class="level3" data-number="7.5.1">
<h3 data-number="7.5.1" class="anchored" data-anchor-id="sec-understanding-pprl-intro"><span class="header-section-number">7.5.1</span> Introduction to PPRL</h3>
<p>Privacy Preserving Record Linkage (PPRL) is a cryptographically-secure method to link information about individuals from different sources of data, without revealing personal information about those individuals. PPRL enables N3C to collect information such as EHR data from clinical data partners, viral variant information from sequencing centers, mortality information from government and private sources, and government Medicare and Medicaid data, all into a single unified database for researcher use.</p>
</section>
<section id="sec-understanding-pprl-access" class="level3" data-number="7.5.2">
<h3 data-number="7.5.2" class="anchored" data-anchor-id="sec-understanding-pprl-access"><span class="header-section-number">7.5.2</span> PPRL Data Access</h3>
<p>Access to datasets that incorporate the PPRL are available through the N3C Data Use Request process. PPRL requests are treated as level 3 / Limited Data Set requests and as such require a signed institutional Data Use Agreement (DUA), a local letter of determination, investigator attestation to the N3C Code of Conduct, IT security training, and human subject training. Requests for use of datasets linked by the PPRL require approval by the National Center for Advancing Translational Sciences (NCATS) federally staffed Data Access Committee (DAC).</p>
</section>
<section id="sec-understanding-pprl-mortality" class="level3" data-number="7.5.3">
<h3 data-number="7.5.3" class="anchored" data-anchor-id="sec-understanding-pprl-mortality"><span class="header-section-number">7.5.3</span> Mortality</h3>
<section id="sec-understanding-pprl-mortality-n3c" class="level4" data-number="7.5.3.1">
<h4 data-number="7.5.3.1" class="anchored" data-anchor-id="sec-understanding-pprl-mortality-n3c"><span class="header-section-number">7.5.3.1</span> Mortality Data Within N3C</h4>
<p>To provide more complete information on mortality, N3C has collected additional mortality data from multiple sources. Because the data sources themselves are sensitive information, supplemental PPRL mortality records are broken down into three source categories:</p>
<ul>
<li><em>Government Mortality</em>: Government data sourced from death certificates and person-reporting.</li>
<li><em>Public Obituary</em>: Obituary data sourced from funeral homes, newspapers, and other online obituary sources specifically from <a href="https://www.obituarydata.com/" class="uri">https://www.obituarydata.com/</a> (a private obituary aggregator).</li>
<li><em>Private Obituary</em>: Obituary data sourced from funeral homes, newspapers, and other online obituary sources sourced from other private sources.</li>
</ul>
<p>PPRL mortality data are harmonized against the existing Level-3 (LDS) OMOP tables in the N3C Data Enclave. Once access is granted via a Level 3 DUR requesting mortality data, PPRL tables may be found in the Data Catalog under the PPRL Datasets collection. The primary mortality data are available in a table simply called mortality.</p>
</section>
<section id="sec-understanding-pprl-mortality-caveats" class="level4" data-number="7.5.3.2">
<h4 data-number="7.5.3.2" class="anchored" data-anchor-id="sec-understanding-pprl-mortality-caveats"><span class="header-section-number">7.5.3.2</span> Mortality Data Completeness and Caveats</h4>
<p>First, note that mortality information is only available for those data partners who have opted in to linkage of their records against the mortality data. As a result the data_partner_id column will only represent a subset of the data partner IDs found in other N3C OMOP tables. See the <a href="https://unite.nih.gov/workspace/notepad/view/ri.notepad.main.notepad.095ada1d-1a0c-48d6-80ba-2d4c547b362c">Intro to PPRL documentation</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> for more information.</p>
<p>Secondly, these data sources should not be considered comprehensive in the sense of providing full information on all deaths in the US. As a result, there may well be mortality records in the OMOP death table that are not represented in the supplemental PPRL data. (And certainly there are mortality records in the PPRL data that are not present in the OMOP death table–that’s why N3C has collected this data in the first place!)</p>
<p>Different data sources lag in inclusion of mortality information from the actual date of death (e.g.&nbsp;if someone’s date_of_death is 2022-04-05, that record may not show up in the mortality source data until 2022-04-18). This lag varies by data source type: government sources tend to lag longer than private sources. For detailed information on mortality data latency, data completeness, and other considerations, see the <a href="https://unite.nih.gov/workspace/notepad/view/ri.notepad.main.notepad.e45cd20e-9da9-44c7-b1d1-747e7c56102f">N3C PPRL Mortality Data Guide</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> and the <a href="https://discovery.biothings.io/faq/n3c#VIRAL_VARIANCE#MORTALITY">N3C Mortality Data FAQs</a>.</p>
</section>
</section>
<section id="sec-understanding-pprl-variants" class="level3" data-number="7.5.4">
<h3 data-number="7.5.4" class="anchored" data-anchor-id="sec-understanding-pprl-variants"><span class="header-section-number">7.5.4</span> Viral Variant</h3>
<section id="sec-understanding-pprl-variants-intro" class="level4" data-number="7.5.4.1">
<h4 data-number="7.5.4.1" class="anchored" data-anchor-id="sec-understanding-pprl-variants-intro"><span class="header-section-number">7.5.4.1</span> What Are Viral Variants?</h4>
<p>Several coronavirus variants have emerged as the virus, SARS-CoV-2, continues to mutate and evolve. For general information about COVID viral variants, see <a href="https://www.hopkinsmedicine.org/health/conditions-and-diseases/coronavirus/a-new-strain-of-coronavirus-what-you-should-know#:~:text=A%20variant%20of%20interest%20is,testing%2C%20or%20more%20severe%20disease.">COVID Variants: What You Should Know</a></p>
</section>
<section id="sec-understanding-pprl-variants-n3c" class="level4" data-number="7.5.4.2">
<h4 data-number="7.5.4.2" class="anchored" data-anchor-id="sec-understanding-pprl-variants-n3c"><span class="header-section-number">7.5.4.2</span> Viral Variants Data Within N3C</h4>
<p>The collection and linkage viral variant PPRL data within N3C is planned in two phases; in the first phase (completed), N3C links patient summary information about sequenced variants. In the second phase (in development), N3C will provide information on the viral variant’s genome sequence via connection to NCBI sequence databases.</p>
<p>For the latest information about viral variant data within N3C, see the N3C <a href="https://unite.nih.gov/workspace/notepad/view/ri.notepad.main.notepad.8514a4bb-b9bb-4b4b-b97a-4e9b1ca61a48">PPRL Viral Variants Guide</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> and <a href="https://discovery.biothings.io/faq/n3c#VIRAL_VARIANCE">N3C Viral Variant FAQs</a>.</p>
</section>
<section id="sec-understanding-pprl-variants-sites" class="level4" data-number="7.5.4.3">
<h4 data-number="7.5.4.3" class="anchored" data-anchor-id="sec-understanding-pprl-variants-sites"><span class="header-section-number">7.5.4.3</span> Which sites</h4>
<p>The initial viral variant data set includes data from a small number of sites. The number of viral variant data sets will increase as more sites from Clinical and Translational Science Awards (CTSA) and Clinical Trials Research (CTR) participate.</p>
</section>
<section id="sec-understanding-pprl-variants-caveats" class="level4" data-number="7.5.4.4">
<h4 data-number="7.5.4.4" class="anchored" data-anchor-id="sec-understanding-pprl-variants-caveats"><span class="header-section-number">7.5.4.4</span> Viral Variants Completeness and Caveats</h4>
<p>The summary data currently available in N3C includes:</p>
<ul>
<li>Sample Date</li>
<li>WHO label (e.g.&nbsp;alpha, delta, omicron)</li>
<li>PANGO lineage (e.g.&nbsp;B.1.1.7)<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a></li>
</ul>
</section>
<section id="sec-understanding-pprl-variants-omop" class="level4" data-number="7.5.4.5">
<h4 data-number="7.5.4.5" class="anchored" data-anchor-id="sec-understanding-pprl-variants-omop"><span class="header-section-number">7.5.4.5</span> Viral Variant Data in OMOP</h4>
<p>Summary PPRL viral variant data is available as an enriched OMOP measurement table available in the PPRL Datasets data. The following columns are used:</p>
<ul>
<li><code>measurement_concept_id</code> contains OMOP <code>concept_id</code> 36033667, “SARS-CoV-2 (COVID-19) variant [Type] in Specimen by Sequencing”, for all viral variant samples.</li>
<li><code>value_as_concept_id</code> will be encoded with the OMOP concept representing the WHO label (e.g.&nbsp;<code>concept_id</code> 4228611 for “Omicron”).</li>
<li>A new column outside the standard OMOP schema includes the PANGO lineage.</li>
</ul>
</section>
</section>
<section id="sec-understanding-pprl-cms" class="level3" data-number="7.5.5">
<h3 data-number="7.5.5" class="anchored" data-anchor-id="sec-understanding-pprl-cms"><span class="header-section-number">7.5.5</span> Centers for Medicare and Medicaid Services (CMS)</h3>
<section id="sec-understanding-pprl-cms-intro" class="level4" data-number="7.5.5.1">
<h4 data-number="7.5.5.1" class="anchored" data-anchor-id="sec-understanding-pprl-cms-intro"><span class="header-section-number">7.5.5.1</span> Introduction to CMS</h4>
<p>The Centers for Medicare and Medicaid Services (CMS) is a federal agency that is part of the U.S. Department of Health and Human Services (HHS) that administers the nation’s Medicare Program. The agency also works with state governments to administer programs that include Medicaid and the Children’s Health Insurance Program (CHIP).</p>
<p><em>Medicare</em> is the federal health insurance program created in 1965 to provide health coverage for Americans aged 65 and older. The program was expanded in 1972 to cover people younger than 65 who have permanent disabilities. More than 64 million Americans are currently covered by Medicare.</p>
<p><em>Medicaid</em> was signed into law in 1965 alongside Medicare. All states, the District of Columbia, and the U.S. territories have Medicaid programs designed to provide health coverage for low-income people. Although the Federal government establishes certain parameters for all states to follow, each state administers their Medicaid program differently, resulting in variations in Medicaid coverage across the country.<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a></p>
<p>For a full description of CMS, see the <a href="https://www.hhs.gov/answers/medicare-and-medicaid/what-is-the-difference-between-medicare-medicaid/index.html">HHS website</a>.</p>
</section>
<section id="sec-understanding-pprl-cms-claims" class="level4" data-number="7.5.5.2">
<h4 data-number="7.5.5.2" class="anchored" data-anchor-id="sec-understanding-pprl-cms-claims"><span class="header-section-number">7.5.5.2</span> CMS Claims Data</h4>
<p>Medical providers (including pharmacies) submit claims for services or drugs for reimbursement of the enrolled participants. The CMS claims provide an audit trail of billing and reimbursement for healthcare services provided to a patient enrolled in an insurance benefit package no matter where the services were rendered. CMS data contain claims from all sites a patient obtained billable services while enrolled and all medications a patient had filled. CMS claims do not contain clinical depth of EHR data and may not contain non-billable (unbillable) services.</p>
</section>
<section id="sec-understanding-pprl-cms-n3c" class="level4" data-number="7.5.5.3">
<h4 data-number="7.5.5.3" class="anchored" data-anchor-id="sec-understanding-pprl-cms-n3c"><span class="header-section-number">7.5.5.3</span> CMS Data in N3C</h4>
<p>The N3C CMS dataset contains de-identified billing data from the Centers for Medicare and Medicaid Services, formatted into the OMOP common data model for N3C researcher use. The initial CMS data within N3C contains over 240K COVID-19 patients and is updated on a monthly basis. The amount of CMS patients available in the Enclave is dependent on sites participating in the NCATS PPRL linkage initiative, and as additional sites implement the tokenization process the number of patients will increase.</p>
<p>CMS records available cover the period from Jan 1, 2017 to the most recently available (three months ago observation period). Although N3C refreshes its CMS data feed on a weekly basis, CMS data has a significant lag, with most records taking six months or more to become available in the system. All seven types of MS claims are included: Part B (PB), Durable Medical Equipment (DME), Inpatient (IP), Outpatient (OP), Skilled Nursing Facility (SNF), Home Health Agency (HHA), Hospice. The records contain masked providers of service and masked care sites linkable to other characteristics.</p>
<p>The Master Beneficiary Summary File (MBSF):</p>
<ul>
<li>Provides enrollment period and reasons for enrollment</li>
<li>Identifies patients with dual enrollment in Medicare and Medicaid</li>
<li>Can be used to distinguish non-users from non-eligible</li>
<li>Identifies periods of continuous enrollment</li>
<li>Can be used to perform annualization</li>
</ul>
</section>
<section id="sec-understanding-pprl-cms-omop" class="level4" data-number="7.5.5.4">
<h4 data-number="7.5.5.4" class="anchored" data-anchor-id="sec-understanding-pprl-cms-omop"><span class="header-section-number">7.5.5.4</span> CMS into OMOP CDM Structure</h4>
<p>The N3C Data Ingestion and Harmonization (DI&amp;H) team worked with partner Acumen to identify CMS data to be ingested. DI&amp;H then created the mapping from native CMS data elements into the OMOP data model. DI&amp;H created a Code Map Service and a Data Transformation Pipeline, which involved combining sometimes up to 45 rows of data into a single patient record and creating constructs, such as coherent visits across claims. The pipeline also required deidentification of patient and provider identifiers.</p>
</section>
<section id="sec-understanding-pprl-cms-caveats" class="level4" data-number="7.5.5.5">
<h4 data-number="7.5.5.5" class="anchored" data-anchor-id="sec-understanding-pprl-cms-caveats"><span class="header-section-number">7.5.5.5</span> CMS Data Completeness and Caveats</h4>
<p>CMS claims data provides slightly different information compared to EHR data; for example, rather than providing medication prescription information, CMS provides information on medication dispensation from the pharmacy. Enrollment in Medicare and Medicaid are well defined and thoroughly tracked, providing insight into patients healthcare utilization vs eligibility that may be lacking in EHR data.</p>
<div id="tbl-understanding-ehr-versus-claims" class="anchored">
<table class="table">
<caption>Table&nbsp;7.2: Differences between EHR and CMS Claims Data</caption>
<colgroup>
<col style="width: 45%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>EHR Data</th>
<th>CMS Claims Data</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><ul>
<li>Usually contains data only about care received in that healthcare system</li>
<li>Reflects orders or intent</li>
<li>Contains deep clinical data (labs, vitals, notes, surveys, history)</li>
<li>May not contain billing data</li>
<li>Usually does not contain eligibility or enrollment</li>
</ul></td>
<td><ul>
<li>Have little clinical data but have data (Dx, Px, etc) from all healthcare sites a patient received services</li>
<li>Reflect performed services</li>
<li>Reflect filled prescriptions</li>
<li>Contain amounts billed and paid</li>
<li>Paid claim indicates enrollment</li>
<li>EHR elements may get altered or added by billing teams or automatically by software to optimize reimbursement (sometimes described as up-coding)</li>
</ul></td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="sec-understanding-pprl-cms-resources" class="level4" data-number="7.5.5.6">
<h4 data-number="7.5.5.6" class="anchored" data-anchor-id="sec-understanding-pprl-cms-resources"><span class="header-section-number">7.5.5.6</span> Additional CMS Resources</h4>
<p>For a detailed investigation of the usefulness of EHR vs administrative claims data, see <span class="citation" data-cites="kharrazi_2017">Kharrazi et al. (<a href="../references.html#ref-kharrazi_2017" role="doc-biblioref">2017</a>)</span>.</p>
<p>For detailed information on the N3C CMS data, see <a href="https://unite.nih.gov/workspace/notepad/view/ri.notepad.main.notepad.899f98b5-2846-41a3-9563-adb75c9042ce">N3C PPRL CMS Data Guide</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>.</p>
<p>CMS Research Data Assistance Center (ResDAC) contains data dictionaries, code books, and enormous amounts of information about Medicare claims and enrollment. It is a good place to figure out what values represent. Look for dictionaries for the <a href="https://www.cms.gov/Research-Statistics-Data-and-Systems/Files-for-Order/LimitedDataSets/StandardAnalyticalFiles">CMS Standard Analytic Files (SAF) or Limited Data Sets (LDS)</a>.</p>
<p>A detailed CMS training webinar is available on <a href="https://www.youtube.com/watch?v=fs0tM7RnL54">YouTube</a>.</p>
</section>
</section>
</section>
<section id="sec-understanding-public" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="sec-understanding-public"><span class="header-section-number">7.6</span> Public/External Datasets</h2>
<p>Because our data are not representative of the geographic locations whence they come, it is important for many analyses to attempt to “correct” the results due to this selection bias. Data sets, called <em>external datasets</em>, are available outside the Enclave that provide information about such locations. There are datasets about the locations themselves (e.g., <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-8BCEA01&amp;view=focus">zip code distances</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>), about the demographics in those locations (e.g., <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-595A20D&amp;view=focus">zip code census</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>), or about covid, in those locations (e.g., <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-EEA964E&amp;view=focus">covid hesitancy by county</a>, <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-B1CD234&amp;view=focus">state policy by date</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>). There are also datasets to help in mapping from zip codes (data available in Level 3 Enclave datasets), such as <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b?id=KO-AA5C326&amp;view=focus">Mapping Zip codes to states and geolocations</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i>. All available datasets are available at <a href="https://discovery.biothings.io/dataset?guide=/guide/n3c/dataset">The Data Discovery Engine</a> (outside the Enclave) and in <a href="https://unite.nih.gov/workspace/module/view/latest/ri.workshop.main.module.3ab34203-d7f3-482e-adbd-f4113bfd1a2b">The Knowledge Store</a> <i class="fa-solid fa-lock" title="Link requires an N3C Enclave account" aria-label="lock"></i> (inside the Enclave; filter on “External Dataset”).</p>
<p>Note that most of these depend on 5-digit zip codes for linking between a patient’s record and the external dataset, and so are useful only in the context of a Level 3 DUR.</p>
<p>Importing such an external dataset entails a process of review that examines the need, the risk to reidentification, and the license. The <a href="https://covid.cd2h.org/external-datasets">External Datasets page</a> gives more detail, including how to request a new dataset.</p>
<p><strong>This concludes an exciting chapter in the never ending story of N3C. To a cinema near you. Real soon.</strong></p>


<div id="refs" class="references csl-bib-body hanging-indent" data-line-spacing="2" role="list" style="display: none">
<div id="ref-bradwell_2022" class="csl-entry" role="listitem">
Bradwell, K. R., Wooldridge, J. T., Amor, B., Bennett, T. D., Anand, A., Bremer, C., Yoo, Y. J., Qian, Z., Johnson, S. G., Pfaff, E. R., et al. (2022). Harmonizing units and values of quantitative data elements in a very large nationally pooled electronic health record (EHR) dataset. <em>Journal of the American Medical Informatics Association</em>, <em>29</em>(7), 1172–1182. <a href="https://doi.org/10.1093/jamia/ocac054">https://doi.org/10.1093/jamia/ocac054</a>
</div>
<div id="ref-gold_2018" class="csl-entry" role="listitem">
Gold, S., Batch, A., McClure, R., Jiang, G., Kharrazi, H., Saripalle, R., Huser, V., Weng, C., Roderer, N., Szarfman, A., et al. (2018). Clinical concept value sets and interoperability in health data analytics. <em>AMIA Annual Symposium Proceedings</em>, <em>2018</em>, 480.
</div>
<div id="ref-gold_2021" class="csl-entry" role="listitem">
Gold, S., Lehmann, H., Schilling, L., &amp; Lutters, W. (2021). Practices, norms, and aspirations regarding the construction, validation, and reuse of code sets in the analysis of real-world data. <em>medRxiv</em>, 2021–2010.
</div>
<div id="ref-kharrazi_2017" class="csl-entry" role="listitem">
Kharrazi, H., Chi, W., Chang, H.-Y., Richards, T. M., Gallagher, J. M., Knudson, S. M., &amp; Weiner, J. P. (2017). Comparing population-based risk-stratification model performance using demographic, diagnosis and medication data extracted from outpatient electronic health records versus administrative claims. <em>Medical Care</em>, <em>55</em>(8), 789–796. <a href="https://doi.org/10.1097/MLR.0000000000000754">https://doi.org/10.1097/MLR.0000000000000754</a>
</div>
<div id="ref-ohdsi_2019" class="csl-entry" role="listitem">
OHDSI. (2019). <em>The book of OHDSI: Observational health data sciences and informatics</em>. OHDSI. <a href="https://ohdsi.github.io/TheBookOfOhdsi/">https://ohdsi.github.io/TheBookOfOhdsi/</a>
</div>
<div id="ref-richesson_2013" class="csl-entry" role="listitem">
Richesson, R. L., Hammond, W. E., Nahm, M., Wixted, D., Simon, G. E., Robinson, J. G., Bauck, A. E., Cifelli, D., Smerek, M. M., Dickerson, J., et al. (2013). Electronic health records based phenotyping in next-generation clinical trials: A perspective from the NIH health care systems collaboratory. <em>Journal of the American Medical Informatics Association</em>, <em>20</em>(e2), e226–e231. <a href="https://doi.org/10.1136/amiajnl-2013-001926">https://doi.org/10.1136/amiajnl-2013-001926</a>
</div>
</div>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>This translation is automated with software, as opposed to manually performed by a human.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>A <em>phenotype</em> is general and a <em>cohort</em> is specific. Thus, a new-onset diabetes phenotype might say, “Must have some number of outpatient visits without a diabetes diagnostic code, followed by at least one visit with such a code.” A related <em>cohort</em> would be, “Must have _2 _outpatient visits without diabetes <em>since Jan 1, 2017</em> without a diabetes diagnostic code, followed by at least one visit with such a code <em>before Jan 1, 2020</em>.” However, in OHDSI the two terms are used interchangeably. See the <a href="https://ohdsi.github.io/TheBookOfOhdsi/Cohorts.html">Cohorts chapter</a> in <span class="citation" data-cites="ohdsi_2019">OHDSI (<a href="../references.html#ref-ohdsi_2019" role="doc-biblioref">2019</a>)</span>.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>Note that while WHO label should always be available, PANGO lineage may only be available for a subset of samples<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>See <a href="https://www.medicareresources.org/glossary/centers-for-medicare-and-medicaid-services/" class="uri">https://www.medicareresources.org/glossary/centers-for-medicare-and-medicaid-services/</a>.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // target, if specified
          link.setAttribute("target", "_blank");
      }
    }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/access.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Getting &amp; Managing Data Access</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/tools.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Introducing Enclave Analysis Tools</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left"><a href="https://ncats.nih.gov/n3c/">N3C</a> © <a href="https://creativecommons.org/licenses/by-nd/4.0/"> CC BY-ND 4</a></div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">Part of <a href="https://covid.cd2h.org/">NIH NCATS</a>.</div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"loop":true,"selector":".lightbox","closeEffect":"zoom","descPosition":"bottom","openEffect":"zoom"});</script>



</body></html>